<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 ASSIGNING TAXONOMY | 16S Microbiome Analysis on a Computing Cluster</title>
  <meta name="description" content="8 ASSIGNING TAXONOMY | 16S Microbiome Analysis on a Computing Cluster" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="8 ASSIGNING TAXONOMY | 16S Microbiome Analysis on a Computing Cluster" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="johannabosch/QIIME2_for_Graham" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 ASSIGNING TAXONOMY | 16S Microbiome Analysis on a Computing Cluster" />
  
  
  

<meta name="author" content="Johanna Bosch" />


<meta name="date" content="2023-10-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="diversity-analyses.html"/>
<link rel="next" href="visualizing-taxonomic-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Comparative 16S Amplicon Analysis using QIIME2 on a computing cluster: Tutorial and Resources</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> <strong>INTRODUCTION</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#a-brief-project-review"><i class="fa fa-check"></i><b>1.1</b> <strong>A brief project review</strong></a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introduction.html"><a href="introduction.html#study-species-and-sites"><i class="fa fa-check"></i><b>1.1.1</b> Study species and sites</a></li>
<li class="chapter" data-level="1.1.2" data-path="introduction.html"><a href="introduction.html#sequencing-library-prep"><i class="fa fa-check"></i><b>1.1.2</b> Sequencing &amp; library prep</a></li>
<li class="chapter" data-level="1.1.3" data-path="introduction.html"><a href="introduction.html#files-after-sequencing"><i class="fa fa-check"></i><b>1.1.3</b> Files after sequencing</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#dna-analysis"><i class="fa fa-check"></i><b>1.2</b> <strong>DNA analysis</strong></a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introduction.html"><a href="introduction.html#the-graham-cluster"><i class="fa fa-check"></i><b>1.2.1</b> The Graham cluster</a></li>
<li class="chapter" data-level="1.2.2" data-path="introduction.html"><a href="introduction.html#data-accessibility"><i class="fa fa-check"></i><b>1.2.2</b> Data accessibility</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html"><i class="fa fa-check"></i><b>2</b> <strong>COMPUTE CANADA BASICS</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#cluster-computing-basics"><i class="fa fa-check"></i><b>2.1</b> <strong>Cluster computing basics</strong></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#signing-into-the-cluster"><i class="fa fa-check"></i><b>2.1.1</b> Signing into the cluster</a></li>
<li class="chapter" data-level="2.1.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#basic-computing-commands"><i class="fa fa-check"></i><b>2.1.2</b> Basic computing commands</a></li>
<li class="chapter" data-level="2.1.3" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#loading-modules"><i class="fa fa-check"></i><b>2.1.3</b> Loading modules</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#job-scripts"><i class="fa fa-check"></i><b>2.2</b> <strong>Job scripts </strong></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#using-nano"><i class="fa fa-check"></i><b>2.2.1</b> Using Nano</a></li>
<li class="chapter" data-level="2.2.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#writing-a-bash-script"><i class="fa fa-check"></i><b>2.2.2</b> Writing a bash script</a></li>
<li class="chapter" data-level="2.2.3" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#running-scripts-on-the-scheduler"><i class="fa fa-check"></i><b>2.2.3</b> Running scripts on the scheduler</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="importing-data.html"><a href="importing-data.html"><i class="fa fa-check"></i><b>3</b> <strong>IMPORTING DATA</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="importing-data.html"><a href="importing-data.html#getting-organized"><i class="fa fa-check"></i><b>3.1</b> <strong>Getting organized</strong></a></li>
<li class="chapter" data-level="3.2" data-path="importing-data.html"><a href="importing-data.html#moving-data-to-the-cluster"><i class="fa fa-check"></i><b>3.2</b> <strong>Moving data to the cluster</strong></a></li>
<li class="chapter" data-level="3.3" data-path="importing-data.html"><a href="importing-data.html#organizing-.fastq-files"><i class="fa fa-check"></i><b>3.3</b> <strong>Organizing .FASTQ files</strong></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="importing-data.html"><a href="importing-data.html#unzip-rawdata-files"><i class="fa fa-check"></i><b>3.3.1</b> Unzip rawdata files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html"><i class="fa fa-check"></i><b>4</b> <strong>QIIME2 ON THE CLUSTER</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#use-qiime2-via-apptainer"><i class="fa fa-check"></i><b>4.1</b> <strong>Use QIIME2 via Apptainer</strong></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#finding-physical-paths"><i class="fa fa-check"></i><b>4.1.1</b> Finding physical paths</a></li>
<li class="chapter" data-level="4.1.2" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#using-apptainer-mounts"><i class="fa fa-check"></i><b>4.1.2</b> Using Apptainer mounts</a></li>
<li class="chapter" data-level="4.1.3" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#getting-help"><i class="fa fa-check"></i><b>4.1.3</b> Getting help</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html"><i class="fa fa-check"></i><b>5</b> <strong>GETTING STARTED WITH QIIME2</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#creating-artifact-files"><i class="fa fa-check"></i><b>5.1</b> <strong>Creating artifact files</strong></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#import-the-.fastq-files"><i class="fa fa-check"></i><b>5.1.1</b> Import the .FASTQ files</a></li>
<li class="chapter" data-level="5.1.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#inspect-the-.qza-files"><i class="fa fa-check"></i><b>5.1.2</b> Inspect the .QZA files</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#visualizing-artifact-files"><i class="fa fa-check"></i><b>5.2</b> <strong>Visualizing artifact files</strong></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#create-a-summary-report"><i class="fa fa-check"></i><b>5.2.1</b> Create a summary report</a></li>
<li class="chapter" data-level="5.2.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#using-qiime2s-web-app"><i class="fa fa-check"></i><b>5.2.2</b> Using QIIME2’s web-app</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-preprocessing.html"><a href="data-preprocessing.html"><i class="fa fa-check"></i><b>6</b> <strong>DATA PREPROCESSING</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#quality-assessment"><i class="fa fa-check"></i><b>6.1</b> <strong>Quality assessment</strong></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#fastqc"><i class="fa fa-check"></i><b>6.1.1</b> FastQC</a></li>
<li class="chapter" data-level="6.1.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#multiqc"><i class="fa fa-check"></i><b>6.1.2</b> MultiQC</a></li>
<li class="chapter" data-level="6.1.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#visualizing-data-quality"><i class="fa fa-check"></i><b>6.1.3</b> Visualizing data quality</a></li>
<li class="chapter" data-level="6.1.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#plot-types"><i class="fa fa-check"></i><b>6.1.4</b> Plot types</a></li>
<li class="chapter" data-level="6.1.5" data-path="data-preprocessing.html"><a href="data-preprocessing.html#for-this-study"><i class="fa fa-check"></i><b>6.1.5</b> For this study</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-cleaning"><i class="fa fa-check"></i><b>6.2</b> <strong>Data Cleaning</strong></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#trimming-primers"><i class="fa fa-check"></i><b>6.2.1</b> Trimming primers</a></li>
<li class="chapter" data-level="6.2.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#summarizing-data"><i class="fa fa-check"></i><b>6.2.2</b> Summarizing data</a></li>
<li class="chapter" data-level="6.2.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#joining-read-pairs"><i class="fa fa-check"></i><b>6.2.3</b> Joining read pairs</a></li>
<li class="chapter" data-level="6.2.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#quality-filtering"><i class="fa fa-check"></i><b>6.2.4</b> Quality filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-transformation"><i class="fa fa-check"></i><b>6.3</b> <strong>Data transformation</strong></a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#constructing-asvs"><i class="fa fa-check"></i><b>6.3.1</b> Constructing ASV’s</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-reduction"><i class="fa fa-check"></i><b>6.4</b> <strong>Data reduction</strong></a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#filtering-unique-reads"><i class="fa fa-check"></i><b>6.4.1</b> Filtering unique reads</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="data-preprocessing.html"><a href="data-preprocessing.html#manipulating-tables"><i class="fa fa-check"></i><b>6.5</b> <strong>Manipulating tables</strong></a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#merging-feature-tables"><i class="fa fa-check"></i><b>6.5.1</b> Merging feature tables</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="diversity-analyses.html"><a href="diversity-analyses.html"><i class="fa fa-check"></i><b>7</b> <strong>DIVERSITY ANALYSES</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#build-a-phylogenetic-tree"><i class="fa fa-check"></i><b>7.1</b> <strong>Build a phylogenetic tree</strong></a></li>
<li class="chapter" data-level="7.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#generating-alpha-rarefaction-curves"><i class="fa fa-check"></i><b>7.2</b> <strong>Generating alpha rarefaction curves</strong></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#finding-max-depth"><i class="fa fa-check"></i><b>7.2.1</b> Finding max depth</a></li>
<li class="chapter" data-level="7.2.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#generating-the-curve"><i class="fa fa-check"></i><b>7.2.2</b> Generating the curve</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-and-beta-diversity"><i class="fa fa-check"></i><b>7.3</b> <strong>Alpha and beta diversity</strong></a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#running-diversity-analyses"><i class="fa fa-check"></i><b>7.3.1</b> Running diversity analyses</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html"><i class="fa fa-check"></i><b>8</b> <strong>ASSIGNING TAXONOMY</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#classifying-asvs"><i class="fa fa-check"></i><b>8.1</b> <strong>Classifying ASV’s</strong></a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#using-scikit-learn"><i class="fa fa-check"></i><b>8.1.1</b> Using scikit-learn</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#verifying-with-blastn"><i class="fa fa-check"></i><b>8.2</b> <strong>Verifying with BLASTn</strong></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#filtering-contaminants"><i class="fa fa-check"></i><b>8.2.1</b> Filtering contaminants</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#grouping-feature-tables"><i class="fa fa-check"></i><b>8.3</b> <strong>Grouping feature tables</strong></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html"><i class="fa fa-check"></i><b>9</b> <strong>VISUALIZING TAXONOMIC DATA</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#exporting-data"><i class="fa fa-check"></i><b>9.1</b> <strong>Exporting data</strong></a></li>
<li class="chapter" data-level="9.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#importing-q2-files-to-r"><i class="fa fa-check"></i><b>9.2</b> <strong>Importing Q2 files to R</strong></a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#qiime2r"><i class="fa fa-check"></i><b>9.3</b> <strong>QIIME2R</strong></a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-feature-table"><i class="fa fa-check"></i><b>9.3.1</b> Load the feature table</a></li>
<li class="chapter" data-level="9.3.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-metadata"><i class="fa fa-check"></i><b>9.3.2</b> Load the metadata</a></li>
<li class="chapter" data-level="9.3.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-taxonomic-data"><i class="fa fa-check"></i><b>9.3.3</b> Load the taxonomic data</a></li>
<li class="chapter" data-level="9.3.4" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#parsing-taxonomic-data"><i class="fa fa-check"></i><b>9.3.4</b> Parsing taxonomic data</a></li>
<li class="chapter" data-level="9.3.5" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#create-a-taxa-barplot"><i class="fa fa-check"></i><b>9.3.5</b> Create a taxa barplot</a></li>
<li class="chapter" data-level="9.3.6" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#create-a-taxa-heatmap"><i class="fa fa-check"></i><b>9.3.6</b> Create a taxa heatmap</a></li>
<li class="chapter" data-level="9.3.7" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#plot-beta-diversity"><i class="fa fa-check"></i><b>9.3.7</b> Plot beta-diversity</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#phyloseq"><i class="fa fa-check"></i><b>9.4</b> <strong>Phyloseq</strong></a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#build-a-phyloseq-object"><i class="fa fa-check"></i><b>9.4.1</b> Build a phyloseq object</a></li>
<li class="chapter" data-level="9.4.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#taxonomic-characteristics"><i class="fa fa-check"></i><b>9.4.2</b> Taxonomic characteristics</a></li>
<li class="chapter" data-level="9.4.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#plot-alpha-diversity"><i class="fa fa-check"></i><b>9.4.3</b> Plot alpha-diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>10</b> <strong>Resources:</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="resources.html"><a href="resources.html#data-in-this-analysis"><i class="fa fa-check"></i><b>10.1</b> <strong>Data in this analysis</strong></a></li>
<li class="chapter" data-level="10.2" data-path="resources.html"><a href="resources.html#other-resources"><i class="fa fa-check"></i><b>10.2</b> <strong>Other resources</strong></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>11</b> Acknowledgments</a></li>
<li class="chapter" data-level="12" data-path="technical-support.html"><a href="technical-support.html"><i class="fa fa-check"></i><b>12</b> <strong>Technical Support</strong></a></li>
<li class="chapter" data-level="13" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>13</b> <strong>References</strong></a>
<ul>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html#references-1"><i class="fa fa-check"></i><strong>References</strong></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">16S Microbiome Analysis on a Computing Cluster</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="assigning-taxonomy" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">8</span> <strong>ASSIGNING TAXONOMY</strong><a href="assigning-taxonomy.html#assigning-taxonomy" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Assigning taxonomy to ASVs (Amplicon Sequence Variants) involves classifying unique genetic sequences into taxonomic categories (e.g., genus or species) to identify the organisms they originate from. This helps researchers understand the microbial composition in a sample based on high-resolution genetic data. To assign taxonomy to ASVs using a reference database, you can use a sequence classification tool like QIIME2’s feature-classifier. In this case, you basically provide the ASV sequences as input, and the tool matches them to known sequences in the reference database to assign taxonomic labels.</p>
<div id="classifying-asvs" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> <strong>Classifying ASV’s</strong><a href="assigning-taxonomy.html#classifying-asvs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There are three options for classifying ASVs taxonomically in QIIME:
1) <code>feature-classifier classify-sklearn</code>
2) <code>feature-classifier consensus-vsearch</code>
3) <code>feature-classifier classify-consensus-vsearch</code></p>
<p>Here, I used the <code>classify-sklearn</code> approach, which is known for its speed and efficiency, and can be advantageous when working with large data sets. It’s also a well-documented and widely used method, making it a reliable option with ample community support. However, it’s important to note that the best choice may depend on your specific data set and research goals, so it’s always a good idea to consider the strengths and limitations of each classification method before making a decision.</p>
<div id="using-scikit-learn" class="section level3 hasAnchor" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Using scikit-learn<a href="assigning-taxonomy.html#using-scikit-learn" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Scikit-learn (sklearn) is a powerful machine learning library, including a naive Bayes classifier, used to categorize marker-gene sequences in QIIME based on a pre-trained model (silva-138-99-nb-classifier.qza), revealing the composition of microbial communities in the data set.</p>
<p>To begin, a taxonomic classifier was obtained from the full length SILVA reference sequences on the Qiime2 website: (<a href="https://docs.qiime2.org/2022.2/data-resources/" class="uri">https://docs.qiime2.org/2022.2/data-resources/</a>)</p>
<p>Download the classifier into the main directory with the <code>wget</code> command (this can take a long time and can also be done in a job script)</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="assigning-taxonomy.html#cb56-1" tabindex="-1"></a>wget https<span class="sc">:</span><span class="er">//</span>data.qiime2.org<span class="sc">/</span><span class="fl">2023.7</span><span class="sc">/</span>common<span class="sc">/</span>silva<span class="dv">-138-99</span><span class="sc">-</span>nb<span class="sc">-</span>classifier.qza</span></code></pre></div>
<p>Next, run the full length 16S taxonomic classification on the read files with the following qiime feature-classifier:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="assigning-taxonomy.html#cb57-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb57-2"><a href="assigning-taxonomy.html#cb57-2" tabindex="-1"></a><span class="co">#SBATCH --time=</span></span>
<span id="cb57-3"><a href="assigning-taxonomy.html#cb57-3" tabindex="-1"></a><span class="co">#SBATCH --account=&lt;group-ID&gt;</span></span>
<span id="cb57-4"><a href="assigning-taxonomy.html#cb57-4" tabindex="-1"></a><span class="co">#SBATCH --mem=</span></span>
<span id="cb57-5"><a href="assigning-taxonomy.html#cb57-5" tabindex="-1"></a></span>
<span id="cb57-6"><a href="assigning-taxonomy.html#cb57-6" tabindex="-1"></a>export APPTAINER_BIND<span class="ot">=</span><span class="er">/</span>project<span class="sc">/</span><span class="dv">6011879</span><span class="sc">/</span><span class="er">&lt;</span>user<span class="sc">-</span>ID<span class="sc">&gt;</span><span class="er">/:/</span>data</span>
<span id="cb57-7"><a href="assigning-taxonomy.html#cb57-7" tabindex="-1"></a>export APPTAINER_WORKDIR<span class="ot">=</span><span class="er">$</span>{SLURM_TMPDIR}</span>
<span id="cb57-8"><a href="assigning-taxonomy.html#cb57-8" tabindex="-1"></a></span>
<span id="cb57-9"><a href="assigning-taxonomy.html#cb57-9" tabindex="-1"></a>module load qiime2<span class="sc">/</span><span class="fl">2023.5</span></span>
<span id="cb57-10"><a href="assigning-taxonomy.html#cb57-10" tabindex="-1"></a></span>
<span id="cb57-11"><a href="assigning-taxonomy.html#cb57-11" tabindex="-1"></a><span class="co">#for sediment core samples</span></span>
<span id="cb57-12"><a href="assigning-taxonomy.html#cb57-12" tabindex="-1"></a>qiime feature<span class="sc">-</span>classifier classify<span class="sc">-</span>sklearn \</span>
<span id="cb57-13"><a href="assigning-taxonomy.html#cb57-13" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>reads <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>rep<span class="sc">-</span>seqs_sediments.qza \</span>
<span id="cb57-14"><a href="assigning-taxonomy.html#cb57-14" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>classifier <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>silva<span class="dv">-138-99</span><span class="sc">-</span>nb<span class="sc">-</span>classifier.qza \</span>
<span id="cb57-15"><a href="assigning-taxonomy.html#cb57-15" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>classification <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>classification_sediments.qza</span>
<span id="cb57-16"><a href="assigning-taxonomy.html#cb57-16" tabindex="-1"></a></span>
<span id="cb57-17"><a href="assigning-taxonomy.html#cb57-17" tabindex="-1"></a><span class="co">#for seabird fecal swab samples:</span></span>
<span id="cb57-18"><a href="assigning-taxonomy.html#cb57-18" tabindex="-1"></a>qiime feature<span class="sc">-</span>classifier classify<span class="sc">-</span>sklearn \</span>
<span id="cb57-19"><a href="assigning-taxonomy.html#cb57-19" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>reads <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>rep<span class="sc">-</span>seqs_birds.qza \</span>
<span id="cb57-20"><a href="assigning-taxonomy.html#cb57-20" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>classifier <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>silva<span class="dv">-138-99</span><span class="sc">-</span>nb<span class="sc">-</span>classifier.qza \</span>
<span id="cb57-21"><a href="assigning-taxonomy.html#cb57-21" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>classification <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>classification_birds.qza</span>
<span id="cb57-22"><a href="assigning-taxonomy.html#cb57-22" tabindex="-1"></a></span>
<span id="cb57-23"><a href="assigning-taxonomy.html#cb57-23" tabindex="-1"></a><span class="co">#for the merged file:</span></span>
<span id="cb57-24"><a href="assigning-taxonomy.html#cb57-24" tabindex="-1"></a>qiime feature<span class="sc">-</span>classifier classify<span class="sc">-</span>sklearn \</span>
<span id="cb57-25"><a href="assigning-taxonomy.html#cb57-25" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>reads <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>merged_rep<span class="sc">-</span>seqs.qza \</span>
<span id="cb57-26"><a href="assigning-taxonomy.html#cb57-26" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>classifier <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>silva<span class="dv">-138-99</span><span class="sc">-</span>nb<span class="sc">-</span>classifier.qza \</span>
<span id="cb57-27"><a href="assigning-taxonomy.html#cb57-27" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>classification <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>classification.qza</span></code></pre></div>
</div>
</div>
<div id="verifying-with-blastn" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> <strong>Verifying with BLASTn</strong><a href="assigning-taxonomy.html#verifying-with-blastn" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As a safety measure, use the feature-table plug-in <strong>[17]</strong> to run a <code>tabulate-seqs</code> <strong>[18]</strong> command that will generate a table with all the Feature ID’s associated to each representative sequence. Using this plug-in, it’s also possible to use BLASTn to evaluate the subset of taxonomic assignments in the taxa files. By comparing the taxonomic assignments from the preceding output to the leading BLASTn hits, you can verify the classifier’s accuracy. Let’s do this for the <code>rep-seqs</code> file for the sediment and seabird fecal samples, as well as the <code>merged_rep-seqs</code> file:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="assigning-taxonomy.html#cb58-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb58-2"><a href="assigning-taxonomy.html#cb58-2" tabindex="-1"></a><span class="co">#SBATCH --time=</span></span>
<span id="cb58-3"><a href="assigning-taxonomy.html#cb58-3" tabindex="-1"></a><span class="co">#SBATCH --account=&lt;group-ID&gt;</span></span>
<span id="cb58-4"><a href="assigning-taxonomy.html#cb58-4" tabindex="-1"></a><span class="co">#SBATCH --mem=</span></span>
<span id="cb58-5"><a href="assigning-taxonomy.html#cb58-5" tabindex="-1"></a></span>
<span id="cb58-6"><a href="assigning-taxonomy.html#cb58-6" tabindex="-1"></a>export APPTAINER_BIND<span class="ot">=</span><span class="er">/</span>project<span class="sc">/</span><span class="dv">6011879</span><span class="sc">/</span><span class="er">&lt;</span>user<span class="sc">-</span>ID<span class="sc">&gt;</span><span class="er">/:/</span>data</span>
<span id="cb58-7"><a href="assigning-taxonomy.html#cb58-7" tabindex="-1"></a>export APPTAINER_WORKDIR<span class="ot">=</span><span class="er">$</span>{SLURM_TMPDIR}</span>
<span id="cb58-8"><a href="assigning-taxonomy.html#cb58-8" tabindex="-1"></a></span>
<span id="cb58-9"><a href="assigning-taxonomy.html#cb58-9" tabindex="-1"></a>module load qiime2<span class="sc">/</span><span class="fl">2023.5</span></span>
<span id="cb58-10"><a href="assigning-taxonomy.html#cb58-10" tabindex="-1"></a></span>
<span id="cb58-11"><a href="assigning-taxonomy.html#cb58-11" tabindex="-1"></a>qiime feature<span class="sc">-</span>table tabulate<span class="sc">-</span>seqs \</span>
<span id="cb58-12"><a href="assigning-taxonomy.html#cb58-12" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>data <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>rep<span class="sc">-</span>seqs_sediments.qza  \</span>
<span id="cb58-13"><a href="assigning-taxonomy.html#cb58-13" tabindex="-1"></a><span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>BLAST_sediments_rep<span class="sc">-</span>seqs.qzv</span>
<span id="cb58-14"><a href="assigning-taxonomy.html#cb58-14" tabindex="-1"></a></span>
<span id="cb58-15"><a href="assigning-taxonomy.html#cb58-15" tabindex="-1"></a>qiime feature<span class="sc">-</span>table tabulate<span class="sc">-</span>seqs \</span>
<span id="cb58-16"><a href="assigning-taxonomy.html#cb58-16" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>data <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>rep<span class="sc">-</span>seqs_birds.qza  \</span>
<span id="cb58-17"><a href="assigning-taxonomy.html#cb58-17" tabindex="-1"></a><span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>BLAST_bird_rep<span class="sc">-</span>seqs.qzv</span>
<span id="cb58-18"><a href="assigning-taxonomy.html#cb58-18" tabindex="-1"></a></span>
<span id="cb58-19"><a href="assigning-taxonomy.html#cb58-19" tabindex="-1"></a>qiime feature<span class="sc">-</span>table tabulate<span class="sc">-</span>seqs \</span>
<span id="cb58-20"><a href="assigning-taxonomy.html#cb58-20" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>data <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>merged_rep<span class="sc">-</span>seqs.qza  \</span>
<span id="cb58-21"><a href="assigning-taxonomy.html#cb58-21" tabindex="-1"></a><span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>BLAST_merged_rep<span class="sc">-</span>seqs.qzv</span></code></pre></div>
<p>Open the Blast files in QIIME2 and there will be three tables that load:</p>
<ol style="list-style-type: decimal">
<li>The sequence length statistics</li>
<li>The seven-number summary of sequence lengths</li>
<li>The sequence table</li>
</ol>
<p>BLAST a sequence against the NCBI nt database by clicking on the linked sequences within the sequence table, which will will open NCBI’s blastn suite. Click on the <code>View Report</code> button and wait for the suite to load the nt data for this sequence (this can take a while). You can do this with around five sequences at random to determine whether or not the results from the taxonomic classification through QIIME2 align with the NCBI database.</p>
<div id="filtering-contaminants" class="section level3 hasAnchor" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Filtering contaminants<a href="assigning-taxonomy.html#filtering-contaminants" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After assigning taxonomy, it’s possible to remove any contaminants or noise in the data by filtering out unwanted assignments based on the taxonomic labels given to ASVs. Here, I remove any mitochondrial and chloroplast 16S sequences by excluding any ASV which contains those terms in its taxonomic label, including any ASV that is unclassified at the phylum level. We omitted the parameter <code>&lt;--p-include p__&gt;</code> because we are studying a poorly characterized environment where there is potential to identify novel phyla.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="assigning-taxonomy.html#cb59-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb59-2"><a href="assigning-taxonomy.html#cb59-2" tabindex="-1"></a><span class="co">#SBATCH --time=</span></span>
<span id="cb59-3"><a href="assigning-taxonomy.html#cb59-3" tabindex="-1"></a><span class="co">#SBATCH --account=&lt;group-ID&gt;</span></span>
<span id="cb59-4"><a href="assigning-taxonomy.html#cb59-4" tabindex="-1"></a><span class="co">#SBATCH --mem=</span></span>
<span id="cb59-5"><a href="assigning-taxonomy.html#cb59-5" tabindex="-1"></a></span>
<span id="cb59-6"><a href="assigning-taxonomy.html#cb59-6" tabindex="-1"></a>export APPTAINER_BIND<span class="ot">=</span><span class="er">/</span>project<span class="sc">/</span><span class="dv">6011879</span><span class="sc">/</span><span class="er">&lt;</span>user<span class="sc">-</span>ID<span class="sc">&gt;</span><span class="er">/:/</span>data</span>
<span id="cb59-7"><a href="assigning-taxonomy.html#cb59-7" tabindex="-1"></a>export APPTAINER_WORKDIR<span class="ot">=</span><span class="er">$</span>{SLURM_TMPDIR}</span>
<span id="cb59-8"><a href="assigning-taxonomy.html#cb59-8" tabindex="-1"></a></span>
<span id="cb59-9"><a href="assigning-taxonomy.html#cb59-9" tabindex="-1"></a>module load qiime2<span class="sc">/</span><span class="fl">2023.5</span></span>
<span id="cb59-10"><a href="assigning-taxonomy.html#cb59-10" tabindex="-1"></a></span>
<span id="cb59-11"><a href="assigning-taxonomy.html#cb59-11" tabindex="-1"></a><span class="co">#for sediment samples</span></span>
<span id="cb59-12"><a href="assigning-taxonomy.html#cb59-12" tabindex="-1"></a>qiime taxa filter<span class="sc">-</span>table \</span>
<span id="cb59-13"><a href="assigning-taxonomy.html#cb59-13" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>filtered_feature<span class="sc">-</span>table_sediments.qza \</span>
<span id="cb59-14"><a href="assigning-taxonomy.html#cb59-14" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>taxonomy <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>classification_sediments.qza \</span>
<span id="cb59-15"><a href="assigning-taxonomy.html#cb59-15" tabindex="-1"></a><span class="sc">--</span>p<span class="sc">-</span>exclude mitochondria,chloroplast \</span>
<span id="cb59-16"><a href="assigning-taxonomy.html#cb59-16" tabindex="-1"></a><span class="sc">--</span>o<span class="sc">-</span>filtered<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>final_table<span class="sc">-</span>sediments.qza</span>
<span id="cb59-17"><a href="assigning-taxonomy.html#cb59-17" tabindex="-1"></a></span>
<span id="cb59-18"><a href="assigning-taxonomy.html#cb59-18" tabindex="-1"></a> qiime feature<span class="sc">-</span>table summarize \</span>
<span id="cb59-19"><a href="assigning-taxonomy.html#cb59-19" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>final<span class="sc">-</span>table_sediments.qza \</span>
<span id="cb59-20"><a href="assigning-taxonomy.html#cb59-20" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>summaries<span class="sc">/</span>final_table<span class="sc">-</span>sediments.qzv</span>
<span id="cb59-21"><a href="assigning-taxonomy.html#cb59-21" tabindex="-1"></a></span>
<span id="cb59-22"><a href="assigning-taxonomy.html#cb59-22" tabindex="-1"></a></span>
<span id="cb59-23"><a href="assigning-taxonomy.html#cb59-23" tabindex="-1"></a><span class="co">#for bird fecal samples</span></span>
<span id="cb59-24"><a href="assigning-taxonomy.html#cb59-24" tabindex="-1"></a>qiime taxa filter<span class="sc">-</span>table \</span>
<span id="cb59-25"><a href="assigning-taxonomy.html#cb59-25" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>filtered_feature<span class="sc">-</span>table_birds.qza \</span>
<span id="cb59-26"><a href="assigning-taxonomy.html#cb59-26" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>taxonomy <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>classification_birds.qza \</span>
<span id="cb59-27"><a href="assigning-taxonomy.html#cb59-27" tabindex="-1"></a><span class="sc">--</span>p<span class="sc">-</span>exclude mitochondria,chloroplast \</span>
<span id="cb59-28"><a href="assigning-taxonomy.html#cb59-28" tabindex="-1"></a><span class="sc">--</span>o<span class="sc">-</span>filtered<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>final_table<span class="sc">-</span>birds.qza</span>
<span id="cb59-29"><a href="assigning-taxonomy.html#cb59-29" tabindex="-1"></a></span>
<span id="cb59-30"><a href="assigning-taxonomy.html#cb59-30" tabindex="-1"></a>qiime feature<span class="sc">-</span>table summarize \</span>
<span id="cb59-31"><a href="assigning-taxonomy.html#cb59-31" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>final<span class="sc">-</span>table_birds.qza \</span>
<span id="cb59-32"><a href="assigning-taxonomy.html#cb59-32" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>summaries<span class="sc">/</span>final_table<span class="sc">-</span>birds.qzv</span>
<span id="cb59-33"><a href="assigning-taxonomy.html#cb59-33" tabindex="-1"></a></span>
<span id="cb59-34"><a href="assigning-taxonomy.html#cb59-34" tabindex="-1"></a><span class="co">#for the merged table</span></span>
<span id="cb59-35"><a href="assigning-taxonomy.html#cb59-35" tabindex="-1"></a>qiime taxa filter<span class="sc">-</span>table \</span>
<span id="cb59-36"><a href="assigning-taxonomy.html#cb59-36" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>merged<span class="sc">-</span>table.qza \</span>
<span id="cb59-37"><a href="assigning-taxonomy.html#cb59-37" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>taxonomy <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>classification.qza \</span>
<span id="cb59-38"><a href="assigning-taxonomy.html#cb59-38" tabindex="-1"></a><span class="sc">--</span>p<span class="sc">-</span>exclude mitochondria,chloroplast \</span>
<span id="cb59-39"><a href="assigning-taxonomy.html#cb59-39" tabindex="-1"></a><span class="sc">--</span>o<span class="sc">-</span>filtered<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>final_merged<span class="sc">-</span>table.qza</span>
<span id="cb59-40"><a href="assigning-taxonomy.html#cb59-40" tabindex="-1"></a></span>
<span id="cb59-41"><a href="assigning-taxonomy.html#cb59-41" tabindex="-1"></a>qiime feature<span class="sc">-</span>table summarize \</span>
<span id="cb59-42"><a href="assigning-taxonomy.html#cb59-42" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>final_merged<span class="sc">-</span>table.qza \</span>
<span id="cb59-43"><a href="assigning-taxonomy.html#cb59-43" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>summaries<span class="sc">/</span>final_merged<span class="sc">-</span>table.qzv</span></code></pre></div>
</div>
</div>
<div id="grouping-feature-tables" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> <strong>Grouping feature tables</strong><a href="assigning-taxonomy.html#grouping-feature-tables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, let’s take the merged feature table and create a grouped feature table that takes the sum of all of the feature counts from each group. This table will be useful later on in the analysis when we are visualizing data. To group all of the samples, you can use the QIIME2 platform to make a feature table that sums up all the feature counts for every featureID according to each sample group with the <code>qiime feature-table group</code> function. This is done using the accompanying <a href="https://github.com/johannabosch/QIIME2_for_Graham/blob/main">metadata.csv</a> files, which specify which samples are a part of which group under the column titled <code>group</code>.</p>
<p>To get a better idea of what is meant by the term ‘grouping’, take some of the data we are using here as an example. In this table below, there are 2 northern gannet (<em>Morus_bassanus</em>) samples and 2 black-legged kittiwake (<em>Rissa_tridactyla</em>) samples. In the metadata file, the column group contains the scientific names for each seabird species, so when you run the grouping command and specifiy this column the feature table will be summed up accordingly. Right now this feature table (assigned to the term <code>bird_features</code>) contains feature counts for every sample individually, but we want it to be grouped, so that it looks like Table 2:</p>
<p><strong>Table 1. Structure of bird_features file:</strong></p>
<table>
<thead>
<tr class="header">
<th>FeatureID</th>
<th>NOGA-01</th>
<th>NOGA-02</th>
<th>BLKI-01</th>
<th>BLKI-02</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>4595dec6ed…</td>
<td>3498</td>
<td>10022</td>
<td>1244</td>
<td>589</td>
</tr>
</tbody>
</table>
<p><strong>Table 2. Structure of a grouped bird_features file:</strong></p>
<table>
<thead>
<tr class="header">
<th>FeatureID</th>
<th>Morus_bassanus</th>
<th>Rissa_tridactyla</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>4595dec6ed…</td>
<td>13520</td>
<td>1833</td>
</tr>
</tbody>
</table>
<p>To group the <code>final_merged-table.qza</code>, use the following command:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="assigning-taxonomy.html#cb60-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb60-2"><a href="assigning-taxonomy.html#cb60-2" tabindex="-1"></a><span class="co">#SBATCH --time=</span></span>
<span id="cb60-3"><a href="assigning-taxonomy.html#cb60-3" tabindex="-1"></a><span class="co">#SBATCH --account=&lt;group-ID&gt;</span></span>
<span id="cb60-4"><a href="assigning-taxonomy.html#cb60-4" tabindex="-1"></a><span class="co">#SBATCH --mem=</span></span>
<span id="cb60-5"><a href="assigning-taxonomy.html#cb60-5" tabindex="-1"></a></span>
<span id="cb60-6"><a href="assigning-taxonomy.html#cb60-6" tabindex="-1"></a>export APPTAINER_BIND<span class="ot">=</span><span class="er">/</span>project<span class="sc">/</span><span class="dv">6011879</span><span class="sc">/</span><span class="er">&lt;</span>user<span class="sc">-</span>ID<span class="sc">&gt;</span><span class="er">/:/</span>data</span>
<span id="cb60-7"><a href="assigning-taxonomy.html#cb60-7" tabindex="-1"></a>export APPTAINER_WORKDIR<span class="ot">=</span><span class="er">$</span>{SLURM_TMPDIR}</span>
<span id="cb60-8"><a href="assigning-taxonomy.html#cb60-8" tabindex="-1"></a></span>
<span id="cb60-9"><a href="assigning-taxonomy.html#cb60-9" tabindex="-1"></a>module load qiime2<span class="sc">/</span><span class="fl">2023.5</span></span>
<span id="cb60-10"><a href="assigning-taxonomy.html#cb60-10" tabindex="-1"></a></span>
<span id="cb60-11"><a href="assigning-taxonomy.html#cb60-11" tabindex="-1"></a>qiime feature<span class="sc">-</span>table group \</span>
<span id="cb60-12"><a href="assigning-taxonomy.html#cb60-12" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>final_merged<span class="sc">-</span>table.qza \</span>
<span id="cb60-13"><a href="assigning-taxonomy.html#cb60-13" tabindex="-1"></a> <span class="sc">--</span>p<span class="sc">-</span>axis sample \</span>
<span id="cb60-14"><a href="assigning-taxonomy.html#cb60-14" tabindex="-1"></a> <span class="sc">--</span>m<span class="sc">-</span>metadata<span class="sc">-</span>file <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>metadata<span class="sc">/</span>metadata.csv \</span>
<span id="cb60-15"><a href="assigning-taxonomy.html#cb60-15" tabindex="-1"></a> <span class="sc">--</span>m<span class="sc">-</span>metadata<span class="sc">-</span>column group \</span>
<span id="cb60-16"><a href="assigning-taxonomy.html#cb60-16" tabindex="-1"></a> <span class="sc">--</span>p<span class="sc">-</span>mode sum \</span>
<span id="cb60-17"><a href="assigning-taxonomy.html#cb60-17" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>grouped<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>grouped<span class="sc">-</span>table.qza</span>
<span id="cb60-18"><a href="assigning-taxonomy.html#cb60-18" tabindex="-1"></a></span>
<span id="cb60-19"><a href="assigning-taxonomy.html#cb60-19" tabindex="-1"></a>qiime feature<span class="sc">-</span>table summarize \</span>
<span id="cb60-20"><a href="assigning-taxonomy.html#cb60-20" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>grouped<span class="sc">-</span>table.qza \</span>
<span id="cb60-21"><a href="assigning-taxonomy.html#cb60-21" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>summaries<span class="sc">/</span>grouped<span class="sc">-</span>table.qzv</span></code></pre></div>
<p>If you’re working the seabird and sediments data, you now have a table that contains all of the data, summed up by each group listed in the <code>metedata-grouped.csv</code> table.
___</p>
 
<hr />
<p style="text-align: center;">
Github: <a href="https://github.com/johannabosch">johannabosch</a> </a>
</p>
<p style="text-align: center;">
<span style="color: #808080;"><em><a href="mailto:yohannabosch@gmail.com" class="email">yohannabosch@gmail.com</a></em></span>
</p>
<!-- Add icon library -->
<p><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"></p>
<!-- Add font awesome icons -->
<p style="text-align: center;">
<a href="https://www.instagram.com/yohannabosch/" class="fa fa-instagram"></a>
<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
<a href="https://github.com/johannabosch/" class="fa fa-github"></a>
</p>
<p> </p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="diversity-analyses.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualizing-taxonomic-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/08-AssigningTaxonomy.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
