<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 DIVERSITY ANALYSES | 16S Microbiome Analysis on a Computing Cluster</title>
  <meta name="description" content="7 DIVERSITY ANALYSES | 16S Microbiome Analysis on a Computing Cluster" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="7 DIVERSITY ANALYSES | 16S Microbiome Analysis on a Computing Cluster" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="johannabosch/QIIME2_for_Graham" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 DIVERSITY ANALYSES | 16S Microbiome Analysis on a Computing Cluster" />
  
  
  

<meta name="author" content="Johanna Bosch" />


<meta name="date" content="2023-10-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-preprocessing.html"/>
<link rel="next" href="assigning-taxonomy.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Comparative 16S Amplicon Analysis using QIIME2 on a computing cluster: Tutorial and Resources</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> <strong>INTRODUCTION</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#a-brief-project-review"><i class="fa fa-check"></i><b>1.1</b> <strong>A brief project review</strong></a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introduction.html"><a href="introduction.html#study-species-and-sites"><i class="fa fa-check"></i><b>1.1.1</b> Study species and sites</a></li>
<li class="chapter" data-level="1.1.2" data-path="introduction.html"><a href="introduction.html#sequencing-library-prep"><i class="fa fa-check"></i><b>1.1.2</b> Sequencing &amp; library prep</a></li>
<li class="chapter" data-level="1.1.3" data-path="introduction.html"><a href="introduction.html#files-after-sequencing"><i class="fa fa-check"></i><b>1.1.3</b> Files after sequencing</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#dna-analysis"><i class="fa fa-check"></i><b>1.2</b> <strong>DNA analysis</strong></a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introduction.html"><a href="introduction.html#the-graham-cluster"><i class="fa fa-check"></i><b>1.2.1</b> The Graham cluster</a></li>
<li class="chapter" data-level="1.2.2" data-path="introduction.html"><a href="introduction.html#data-accessibility"><i class="fa fa-check"></i><b>1.2.2</b> Data accessibility</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html"><i class="fa fa-check"></i><b>2</b> <strong>COMPUTE CANADA BASICS</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#cluster-computing-basics"><i class="fa fa-check"></i><b>2.1</b> <strong>Cluster computing basics</strong></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#signing-into-the-cluster"><i class="fa fa-check"></i><b>2.1.1</b> Signing into the cluster</a></li>
<li class="chapter" data-level="2.1.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#basic-computing-commands"><i class="fa fa-check"></i><b>2.1.2</b> Basic computing commands</a></li>
<li class="chapter" data-level="2.1.3" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#loading-modules"><i class="fa fa-check"></i><b>2.1.3</b> Loading modules</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#job-scripts"><i class="fa fa-check"></i><b>2.2</b> <strong>Job scripts </strong></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#using-nano"><i class="fa fa-check"></i><b>2.2.1</b> Using Nano</a></li>
<li class="chapter" data-level="2.2.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#writing-a-bash-script"><i class="fa fa-check"></i><b>2.2.2</b> Writing a bash script</a></li>
<li class="chapter" data-level="2.2.3" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#running-scripts-on-the-scheduler"><i class="fa fa-check"></i><b>2.2.3</b> Running scripts on the scheduler</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="importing-data.html"><a href="importing-data.html"><i class="fa fa-check"></i><b>3</b> <strong>IMPORTING DATA</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="importing-data.html"><a href="importing-data.html#getting-organized"><i class="fa fa-check"></i><b>3.1</b> <strong>Getting organized</strong></a></li>
<li class="chapter" data-level="3.2" data-path="importing-data.html"><a href="importing-data.html#moving-data-to-the-cluster"><i class="fa fa-check"></i><b>3.2</b> <strong>Moving data to the cluster</strong></a></li>
<li class="chapter" data-level="3.3" data-path="importing-data.html"><a href="importing-data.html#organizing-.fastq-files"><i class="fa fa-check"></i><b>3.3</b> <strong>Organizing .FASTQ files</strong></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="importing-data.html"><a href="importing-data.html#unzip-rawdata-files"><i class="fa fa-check"></i><b>3.3.1</b> Unzip rawdata files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html"><i class="fa fa-check"></i><b>4</b> <strong>QIIME2 ON THE CLUSTER</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#use-qiime2-via-apptainer"><i class="fa fa-check"></i><b>4.1</b> <strong>Use QIIME2 via Apptainer</strong></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#finding-physical-paths"><i class="fa fa-check"></i><b>4.1.1</b> Finding physical paths</a></li>
<li class="chapter" data-level="4.1.2" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#using-apptainer-mounts"><i class="fa fa-check"></i><b>4.1.2</b> Using Apptainer mounts</a></li>
<li class="chapter" data-level="4.1.3" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#getting-help"><i class="fa fa-check"></i><b>4.1.3</b> Getting help</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html"><i class="fa fa-check"></i><b>5</b> <strong>GETTING STARTED WITH QIIME2</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#creating-artifact-files"><i class="fa fa-check"></i><b>5.1</b> <strong>Creating artifact files</strong></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#import-the-.fastq-files"><i class="fa fa-check"></i><b>5.1.1</b> Import the .FASTQ files</a></li>
<li class="chapter" data-level="5.1.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#inspect-the-.qza-files"><i class="fa fa-check"></i><b>5.1.2</b> Inspect the .QZA files</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#visualizing-artifact-files"><i class="fa fa-check"></i><b>5.2</b> <strong>Visualizing artifact files</strong></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#create-a-summary-report"><i class="fa fa-check"></i><b>5.2.1</b> Create a summary report</a></li>
<li class="chapter" data-level="5.2.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#using-qiime2s-web-app"><i class="fa fa-check"></i><b>5.2.2</b> Using QIIME2’s web-app</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-preprocessing.html"><a href="data-preprocessing.html"><i class="fa fa-check"></i><b>6</b> <strong>DATA PREPROCESSING</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#quality-assessment"><i class="fa fa-check"></i><b>6.1</b> <strong>Quality assessment</strong></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#fastqc"><i class="fa fa-check"></i><b>6.1.1</b> FastQC</a></li>
<li class="chapter" data-level="6.1.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#multiqc"><i class="fa fa-check"></i><b>6.1.2</b> MultiQC</a></li>
<li class="chapter" data-level="6.1.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#visualizing-data-quality"><i class="fa fa-check"></i><b>6.1.3</b> Visualizing data quality</a></li>
<li class="chapter" data-level="6.1.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#plot-types"><i class="fa fa-check"></i><b>6.1.4</b> Plot types</a></li>
<li class="chapter" data-level="6.1.5" data-path="data-preprocessing.html"><a href="data-preprocessing.html#for-this-study"><i class="fa fa-check"></i><b>6.1.5</b> For this study</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-cleaning"><i class="fa fa-check"></i><b>6.2</b> <strong>Data Cleaning</strong></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#trimming-primers"><i class="fa fa-check"></i><b>6.2.1</b> Trimming primers</a></li>
<li class="chapter" data-level="6.2.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#summarizing-data"><i class="fa fa-check"></i><b>6.2.2</b> Summarizing data</a></li>
<li class="chapter" data-level="6.2.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#joining-read-pairs"><i class="fa fa-check"></i><b>6.2.3</b> Joining read pairs</a></li>
<li class="chapter" data-level="6.2.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#quality-filtering"><i class="fa fa-check"></i><b>6.2.4</b> Quality filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-transformation"><i class="fa fa-check"></i><b>6.3</b> <strong>Data transformation</strong></a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#constructing-asvs"><i class="fa fa-check"></i><b>6.3.1</b> Constructing ASV’s</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-reduction"><i class="fa fa-check"></i><b>6.4</b> <strong>Data reduction</strong></a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#filtering-unique-reads"><i class="fa fa-check"></i><b>6.4.1</b> Filtering unique reads</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="data-preprocessing.html"><a href="data-preprocessing.html#manipulating-tables"><i class="fa fa-check"></i><b>6.5</b> <strong>Manipulating tables</strong></a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#merging-feature-tables"><i class="fa fa-check"></i><b>6.5.1</b> Merging feature tables</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="diversity-analyses.html"><a href="diversity-analyses.html"><i class="fa fa-check"></i><b>7</b> <strong>DIVERSITY ANALYSES</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#build-a-phylogenetic-tree"><i class="fa fa-check"></i><b>7.1</b> <strong>Build a phylogenetic tree</strong></a></li>
<li class="chapter" data-level="7.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#generating-alpha-rarefaction-curves"><i class="fa fa-check"></i><b>7.2</b> <strong>Generating alpha rarefaction curves</strong></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#finding-max-depth"><i class="fa fa-check"></i><b>7.2.1</b> Finding max depth</a></li>
<li class="chapter" data-level="7.2.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#generating-the-curve"><i class="fa fa-check"></i><b>7.2.2</b> Generating the curve</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-and-beta-diversity"><i class="fa fa-check"></i><b>7.3</b> <strong>Alpha and beta diversity</strong></a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#running-diversity-analyses"><i class="fa fa-check"></i><b>7.3.1</b> Running diversity analyses</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html"><i class="fa fa-check"></i><b>8</b> <strong>ASSIGNING TAXONOMY</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#classifying-asvs"><i class="fa fa-check"></i><b>8.1</b> <strong>Classifying ASV’s</strong></a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#using-scikit-learn"><i class="fa fa-check"></i><b>8.1.1</b> Using scikit-learn</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#verifying-with-blastn"><i class="fa fa-check"></i><b>8.2</b> <strong>Verifying with BLASTn</strong></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#filtering-contaminants"><i class="fa fa-check"></i><b>8.2.1</b> Filtering contaminants</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#grouping-feature-tables"><i class="fa fa-check"></i><b>8.3</b> <strong>Grouping feature tables</strong></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html"><i class="fa fa-check"></i><b>9</b> <strong>VISUALIZING TAXONOMIC DATA</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#exporting-data"><i class="fa fa-check"></i><b>9.1</b> <strong>Exporting data</strong></a></li>
<li class="chapter" data-level="9.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#importing-q2-files-to-r"><i class="fa fa-check"></i><b>9.2</b> <strong>Importing Q2 files to R</strong></a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#qiime2r"><i class="fa fa-check"></i><b>9.3</b> <strong>QIIME2R</strong></a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-feature-table"><i class="fa fa-check"></i><b>9.3.1</b> Load the feature table</a></li>
<li class="chapter" data-level="9.3.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-metadata"><i class="fa fa-check"></i><b>9.3.2</b> Load the metadata</a></li>
<li class="chapter" data-level="9.3.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-taxonomic-data"><i class="fa fa-check"></i><b>9.3.3</b> Load the taxonomic data</a></li>
<li class="chapter" data-level="9.3.4" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#parsing-taxonomic-data"><i class="fa fa-check"></i><b>9.3.4</b> Parsing taxonomic data</a></li>
<li class="chapter" data-level="9.3.5" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#create-a-taxa-barplot"><i class="fa fa-check"></i><b>9.3.5</b> Create a taxa barplot</a></li>
<li class="chapter" data-level="9.3.6" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#create-a-taxa-heatmap"><i class="fa fa-check"></i><b>9.3.6</b> Create a taxa heatmap</a></li>
<li class="chapter" data-level="9.3.7" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#plot-beta-diversity"><i class="fa fa-check"></i><b>9.3.7</b> Plot beta-diversity</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#phyloseq"><i class="fa fa-check"></i><b>9.4</b> <strong>Phyloseq</strong></a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#build-a-phyloseq-object"><i class="fa fa-check"></i><b>9.4.1</b> Build a phyloseq object</a></li>
<li class="chapter" data-level="9.4.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#taxonomic-characteristics"><i class="fa fa-check"></i><b>9.4.2</b> Taxonomic characteristics</a></li>
<li class="chapter" data-level="9.4.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#plot-alpha-diversity"><i class="fa fa-check"></i><b>9.4.3</b> Plot alpha-diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>10</b> <strong>Resources:</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="resources.html"><a href="resources.html#data-in-this-analysis"><i class="fa fa-check"></i><b>10.1</b> <strong>Data in this analysis</strong></a></li>
<li class="chapter" data-level="10.2" data-path="resources.html"><a href="resources.html#other-resources"><i class="fa fa-check"></i><b>10.2</b> <strong>Other resources</strong></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>11</b> Acknowledgments</a></li>
<li class="chapter" data-level="12" data-path="technical-support.html"><a href="technical-support.html"><i class="fa fa-check"></i><b>12</b> <strong>Technical Support</strong></a></li>
<li class="chapter" data-level="13" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>13</b> <strong>References</strong></a>
<ul>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html#references-1"><i class="fa fa-check"></i><strong>References</strong></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">16S Microbiome Analysis on a Computing Cluster</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="diversity-analyses" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> <strong>DIVERSITY ANALYSES</strong><a href="diversity-analyses.html#diversity-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="build-a-phylogenetic-tree" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> <strong>Build a phylogenetic tree</strong><a href="diversity-analyses.html#build-a-phylogenetic-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>While a phylogenetic tree is not technically used to assess diversity within samples, the files generated in this step are used to assess beta-diversity down below.</p>
<p>To generate the trees, I use the QIIME2 plug-in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5904434//">SEPP</a> <strong>[13-16]</strong>. First, acquire the SEPP reference database for 16S data, and store it in the main working directory (<code>amplicon-analysis</code>). You can find the latest versions on the <a href="https://docs.qiime2.org/2023.7/data-resources/">QIIME2 Docs Data Resources page</a></p>
<p>Instead of downloading the database on your local computer and transferring it to the cluster, just copy the link to the reference database you’d like to use, and then use the <code>wget</code> command to download the proper file:</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="diversity-analyses.html#cb51-1" tabindex="-1"></a>wget https<span class="sc">:</span><span class="er">//</span>data.qiime2.org<span class="sc">/</span><span class="fl">2023.7</span><span class="sc">/</span>common<span class="sc">/</span>sepp<span class="sc">-</span>refs<span class="sc">-</span>silva<span class="fl">-128.</span>qza</span></code></pre></div>
<p>Then use the QIIME2 fragment-insertion command with the SEPP reference database to generate the tree:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="diversity-analyses.html#cb52-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb52-2"><a href="diversity-analyses.html#cb52-2" tabindex="-1"></a><span class="co">#SBATCH --time=</span></span>
<span id="cb52-3"><a href="diversity-analyses.html#cb52-3" tabindex="-1"></a><span class="co">#SBATCH --account=&lt;group-ID&gt;</span></span>
<span id="cb52-4"><a href="diversity-analyses.html#cb52-4" tabindex="-1"></a><span class="co">#SBATCH --mem=</span></span>
<span id="cb52-5"><a href="diversity-analyses.html#cb52-5" tabindex="-1"></a></span>
<span id="cb52-6"><a href="diversity-analyses.html#cb52-6" tabindex="-1"></a>export APPTAINER_BIND<span class="ot">=</span><span class="er">/</span>project<span class="sc">/</span><span class="dv">6011879</span><span class="sc">/</span><span class="er">&lt;</span>user<span class="sc">-</span>ID<span class="sc">&gt;</span><span class="er">/:/</span>data</span>
<span id="cb52-7"><a href="diversity-analyses.html#cb52-7" tabindex="-1"></a>export APPTAINER_WORKDIR<span class="ot">=</span><span class="er">$</span>{SLURM_TMPDIR}</span>
<span id="cb52-8"><a href="diversity-analyses.html#cb52-8" tabindex="-1"></a></span>
<span id="cb52-9"><a href="diversity-analyses.html#cb52-9" tabindex="-1"></a>module load qiime2<span class="sc">/</span><span class="fl">2023.5</span></span>
<span id="cb52-10"><a href="diversity-analyses.html#cb52-10" tabindex="-1"></a></span>
<span id="cb52-11"><a href="diversity-analyses.html#cb52-11" tabindex="-1"></a><span class="co">#build a tree for seabird fecal samples</span></span>
<span id="cb52-12"><a href="diversity-analyses.html#cb52-12" tabindex="-1"></a>qiime fragment<span class="sc">-</span>insertion sepp \</span>
<span id="cb52-13"><a href="diversity-analyses.html#cb52-13" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>representative<span class="sc">-</span>sequences <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>rep<span class="sc">-</span>seqs_birds.qza \</span>
<span id="cb52-14"><a href="diversity-analyses.html#cb52-14" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>reference<span class="sc">-</span>database <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>sepp<span class="sc">-</span>refs<span class="sc">-</span>silva<span class="fl">-128.</span>qza \</span>
<span id="cb52-15"><a href="diversity-analyses.html#cb52-15" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>tree <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>asvs<span class="sc">-</span>tree_birds.qza \</span>
<span id="cb52-16"><a href="diversity-analyses.html#cb52-16" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>placements <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>insertion<span class="sc">-</span>placements_birds.qza</span>
<span id="cb52-17"><a href="diversity-analyses.html#cb52-17" tabindex="-1"></a></span>
<span id="cb52-18"><a href="diversity-analyses.html#cb52-18" tabindex="-1"></a></span>
<span id="cb52-19"><a href="diversity-analyses.html#cb52-19" tabindex="-1"></a><span class="co">#build a tree for sediment samples</span></span>
<span id="cb52-20"><a href="diversity-analyses.html#cb52-20" tabindex="-1"></a>qiime fragment<span class="sc">-</span>insertion sepp \</span>
<span id="cb52-21"><a href="diversity-analyses.html#cb52-21" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>representative<span class="sc">-</span>sequences <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>rep<span class="sc">-</span>seqs_sediments.qza \</span>
<span id="cb52-22"><a href="diversity-analyses.html#cb52-22" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>reference<span class="sc">-</span>database <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>sepp<span class="sc">-</span>refs<span class="sc">-</span>silva<span class="fl">-128.</span>qza \</span>
<span id="cb52-23"><a href="diversity-analyses.html#cb52-23" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>tree <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>asvs<span class="sc">-</span>tree_sediments.qza \</span>
<span id="cb52-24"><a href="diversity-analyses.html#cb52-24" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>placements <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>insertion<span class="sc">-</span>placements_sediments.qza</span>
<span id="cb52-25"><a href="diversity-analyses.html#cb52-25" tabindex="-1"></a></span>
<span id="cb52-26"><a href="diversity-analyses.html#cb52-26" tabindex="-1"></a></span>
<span id="cb52-27"><a href="diversity-analyses.html#cb52-27" tabindex="-1"></a><span class="co">#build a tree for the merged feature table</span></span>
<span id="cb52-28"><a href="diversity-analyses.html#cb52-28" tabindex="-1"></a>qiime fragment<span class="sc">-</span>insertion sepp \</span>
<span id="cb52-29"><a href="diversity-analyses.html#cb52-29" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>representative<span class="sc">-</span>sequences <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>merged_rep<span class="sc">-</span>seqs.qza \</span>
<span id="cb52-30"><a href="diversity-analyses.html#cb52-30" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>reference<span class="sc">-</span>database <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>sepp<span class="sc">-</span>refs<span class="sc">-</span>silva<span class="fl">-128.</span>qza \</span>
<span id="cb52-31"><a href="diversity-analyses.html#cb52-31" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>tree <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>asvs<span class="sc">-</span>tree.qza \</span>
<span id="cb52-32"><a href="diversity-analyses.html#cb52-32" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>placements <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>insertion<span class="sc">-</span>placements.qza</span></code></pre></div>
</div>
<div id="generating-alpha-rarefaction-curves" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> <strong>Generating alpha rarefaction curves</strong><a href="diversity-analyses.html#generating-alpha-rarefaction-curves" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>An alpha rarefaction curve is used for understanding the richness and diversity of microbial communities within a given sample or data set. In the context of microbiome analysis, it provides insights into the number of unique species, known as alpha diversity, at different sampling depths. These curves are generated by repeatedly subsampling the sequencing data at various depths, and measuring diversity metrics (e.g., Shannon or Simpson index) at each step. As the sequencing depth increases, the curve typically plateaus, revealing the point at which additional sequencing data no longer significantly contributes to the detection of new microbial taxa. Alpha rarefaction curves not only help researchers identify optimal sequencing depths but also offer a glimpse into the complex and unique microbial ecosystems present in various environments, making them a fundamental tool in microbiome research.</p>
<p>Start by making a directory called <code>/diversity</code> to keep all the files that will be generated in this next section.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="diversity-analyses.html#cb53-1" tabindex="-1"></a>mkdir diversity</span></code></pre></div>
<div id="finding-max-depth" class="section level3 hasAnchor" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Finding max depth<a href="diversity-analyses.html#finding-max-depth" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To create an alpha-rarefaction curve, first find out what you want the maximum depth (<code>--p-max-depth</code> in the below command) value to be for your data. This might depend on the quality of the data, or your research question. To determine the appropriate value, you should ideally choose a depth that retains a substantial portion of your data set while still ensuring meaningful results. You can start by looking at your filtered feature table to find the maximum sequencing depth present, or you can set it to a value slightly lower to provide a buffer.</p>
<p>I chose maximum depth based on the maximum sequencing depth for both sample types found within the summary files that were generated in the last steps of the Data Preprocessing stage: <code>filtered_feature-table_birds.qzv</code> and <code>filtered_feature-table_sediments.qzv</code>. When opening these files in <a href="https://view.qiime2.org/">QIIME2 View</a>, this number can be found on the first tab of the visualization file, called <code>Overview</code>, or on the <code>Interactive Sample Detail</code> page.</p>
<p>For example, the maximum depth of the bird fecal samples after filtering was 199 289, and the maximum depth for the sediment samples was 33 503.</p>
<p><img src="images/maximumdepth.png" width="85%" /></p>
</div>
<div id="generating-the-curve" class="section level3 hasAnchor" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Generating the curve<a href="diversity-analyses.html#generating-the-curve" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The values found above for the maximum sampling depths can be used for –p-max-depth in the following command:</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="diversity-analyses.html#cb54-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb54-2"><a href="diversity-analyses.html#cb54-2" tabindex="-1"></a><span class="co">#SBATCH --time=</span></span>
<span id="cb54-3"><a href="diversity-analyses.html#cb54-3" tabindex="-1"></a><span class="co">#SBATCH --account=&lt;group-ID&gt;</span></span>
<span id="cb54-4"><a href="diversity-analyses.html#cb54-4" tabindex="-1"></a><span class="co">#SBATCH --mem=</span></span>
<span id="cb54-5"><a href="diversity-analyses.html#cb54-5" tabindex="-1"></a></span>
<span id="cb54-6"><a href="diversity-analyses.html#cb54-6" tabindex="-1"></a>export APPTAINER_BIND<span class="ot">=</span><span class="er">/</span>project<span class="sc">/</span><span class="dv">6011879</span><span class="sc">/</span><span class="er">&lt;</span>user<span class="sc">-</span>ID<span class="sc">&gt;</span><span class="er">/:/</span>data</span>
<span id="cb54-7"><a href="diversity-analyses.html#cb54-7" tabindex="-1"></a>export APPTAINER_WORKDIR<span class="ot">=</span><span class="er">$</span>{SLURM_TMPDIR}</span>
<span id="cb54-8"><a href="diversity-analyses.html#cb54-8" tabindex="-1"></a></span>
<span id="cb54-9"><a href="diversity-analyses.html#cb54-9" tabindex="-1"></a>module load qiime2<span class="sc">/</span><span class="fl">2023.5</span></span>
<span id="cb54-10"><a href="diversity-analyses.html#cb54-10" tabindex="-1"></a></span>
<span id="cb54-11"><a href="diversity-analyses.html#cb54-11" tabindex="-1"></a>qiime diversity alpha<span class="sc">-</span>rarefaction \</span>
<span id="cb54-12"><a href="diversity-analyses.html#cb54-12" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>filtered_feature<span class="sc">-</span>table_sediments.qza \</span>
<span id="cb54-13"><a href="diversity-analyses.html#cb54-13" tabindex="-1"></a> <span class="sc">--</span>p<span class="sc">-</span>max<span class="sc">-</span>depth <span class="dv">365593</span> \</span>
<span id="cb54-14"><a href="diversity-analyses.html#cb54-14" tabindex="-1"></a> <span class="sc">--</span>p<span class="sc">-</span>steps <span class="dv">20</span> \</span>
<span id="cb54-15"><a href="diversity-analyses.html#cb54-15" tabindex="-1"></a> <span class="sc">--</span>p<span class="sc">-</span>metrics <span class="st">&#39;observed_features&#39;</span> \</span>
<span id="cb54-16"><a href="diversity-analyses.html#cb54-16" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>diversity<span class="sc">/</span>rarefaction_sediments.qzv</span>
<span id="cb54-17"><a href="diversity-analyses.html#cb54-17" tabindex="-1"></a></span>
<span id="cb54-18"><a href="diversity-analyses.html#cb54-18" tabindex="-1"></a></span>
<span id="cb54-19"><a href="diversity-analyses.html#cb54-19" tabindex="-1"></a>qiime diversity alpha<span class="sc">-</span>rarefaction \</span>
<span id="cb54-20"><a href="diversity-analyses.html#cb54-20" tabindex="-1"></a> <span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>filtered_feature<span class="sc">-</span>table_birds.qza \</span>
<span id="cb54-21"><a href="diversity-analyses.html#cb54-21" tabindex="-1"></a> <span class="sc">--</span>p<span class="sc">-</span>max<span class="sc">-</span>depth <span class="dv">42252</span> \</span>
<span id="cb54-22"><a href="diversity-analyses.html#cb54-22" tabindex="-1"></a> <span class="sc">--</span>p<span class="sc">-</span>steps <span class="dv">20</span> \</span>
<span id="cb54-23"><a href="diversity-analyses.html#cb54-23" tabindex="-1"></a> <span class="sc">--</span>p<span class="sc">-</span>metrics <span class="st">&#39;observed_features&#39;</span> \</span>
<span id="cb54-24"><a href="diversity-analyses.html#cb54-24" tabindex="-1"></a> <span class="sc">--</span>o<span class="sc">-</span>visualization <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>diversity<span class="sc">/</span>rarefaction_birds.qzv</span></code></pre></div>
</div>
</div>
<div id="alpha-and-beta-diversity" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> <strong>Alpha and beta diversity</strong><a href="diversity-analyses.html#alpha-and-beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In microbiome analysis, alpha diversity is like looking at the diversity within a single sample or community, telling you how many different types of microbes are present and how evenly they’re distributed. On the other hand, beta diversity is about comparing multiple samples to see how they differ in terms of their microbial composition, helping you understand which microbes are unique to each sample and which are shared. So, alpha diversity focuses on diversity within, while beta diversity looks at differences between different microbial communities.</p>
<div id="running-diversity-analyses" class="section level3 hasAnchor" number="7.3.1">
<h3><span class="header-section-number">7.3.1</span> Running diversity analyses<a href="diversity-analyses.html#running-diversity-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>qiime diversity core-metrics-phylogenetic</code> command can be used to run a suite of diversity analyses on all of the samples in your set all at once, and it will also rarefy all samples to the same sample sequencing depth prior to calculating metrics. Let’s run the <code>qiime diversity core-metrics-phylogenetic</code> command on (1) the seabird fecal samples, (2) the sediment core samples, and (3) the merged feature table that was created in the last steps of Ch-06 (Data Preprocessing).</p>
<p>In this command, <code>--p-sampling-depth X</code> represents the depth at which you would like the analysis to be performed at. To retain as many of the samples as possible, I chose the lowest reasonable sample depth taken from the two following summary visualization files: <code>filtered_feature-table_birds.qzv</code> and <code>filtered_feature-table_sediments.qzv</code>. For this part of the analysis a metadata file and phylogenetic tree are also necessary. To view an example of a Q2 metadata file, go back to Ch-01 (Introduction).</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="diversity-analyses.html#cb55-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb55-2"><a href="diversity-analyses.html#cb55-2" tabindex="-1"></a><span class="co">#SBATCH --time=</span></span>
<span id="cb55-3"><a href="diversity-analyses.html#cb55-3" tabindex="-1"></a><span class="co">#SBATCH --account=&lt;group-ID&gt;</span></span>
<span id="cb55-4"><a href="diversity-analyses.html#cb55-4" tabindex="-1"></a><span class="co">#SBATCH --mem=</span></span>
<span id="cb55-5"><a href="diversity-analyses.html#cb55-5" tabindex="-1"></a></span>
<span id="cb55-6"><a href="diversity-analyses.html#cb55-6" tabindex="-1"></a>export APPTAINER_BIND<span class="ot">=</span><span class="er">/</span>project<span class="sc">/</span><span class="dv">6011879</span><span class="sc">/</span><span class="er">&lt;</span>user<span class="sc">-</span>ID<span class="sc">&gt;</span><span class="er">/:/</span>data</span>
<span id="cb55-7"><a href="diversity-analyses.html#cb55-7" tabindex="-1"></a>export APPTAINER_WORKDIR<span class="ot">=</span><span class="er">$</span>{SLURM_TMPDIR}</span>
<span id="cb55-8"><a href="diversity-analyses.html#cb55-8" tabindex="-1"></a></span>
<span id="cb55-9"><a href="diversity-analyses.html#cb55-9" tabindex="-1"></a>module load qiime2<span class="sc">/</span><span class="fl">2023.5</span></span>
<span id="cb55-10"><a href="diversity-analyses.html#cb55-10" tabindex="-1"></a></span>
<span id="cb55-11"><a href="diversity-analyses.html#cb55-11" tabindex="-1"></a><span class="co">#core metrics for seabird fecal sample </span></span>
<span id="cb55-12"><a href="diversity-analyses.html#cb55-12" tabindex="-1"></a>qiime diversity core<span class="sc">-</span>metrics<span class="sc">-</span>phylogenetic \</span>
<span id="cb55-13"><a href="diversity-analyses.html#cb55-13" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>filtered_feature<span class="sc">-</span>table_birds.qza \</span>
<span id="cb55-14"><a href="diversity-analyses.html#cb55-14" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>phylogeny <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>asvs<span class="sc">-</span>tree_birds.qza \</span>
<span id="cb55-15"><a href="diversity-analyses.html#cb55-15" tabindex="-1"></a><span class="sc">--</span>p<span class="sc">-</span>sampling<span class="sc">-</span>depth <span class="dv">42252</span> \</span>
<span id="cb55-16"><a href="diversity-analyses.html#cb55-16" tabindex="-1"></a><span class="sc">--</span>m<span class="sc">-</span>metadata<span class="sc">-</span>file <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>metadata<span class="sc">/</span>metadata<span class="sc">-</span>birds.csv \</span>
<span id="cb55-17"><a href="diversity-analyses.html#cb55-17" tabindex="-1"></a><span class="sc">--</span>output<span class="sc">-</span>dir <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>diversity<span class="sc">/</span>diversity_birds</span>
<span id="cb55-18"><a href="diversity-analyses.html#cb55-18" tabindex="-1"></a></span>
<span id="cb55-19"><a href="diversity-analyses.html#cb55-19" tabindex="-1"></a><span class="co">#core metrics for sediment samples  </span></span>
<span id="cb55-20"><a href="diversity-analyses.html#cb55-20" tabindex="-1"></a>qiime diversity <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>core<span class="sc">-</span>metrics<span class="sc">-</span>phylogenetic \</span>
<span id="cb55-21"><a href="diversity-analyses.html#cb55-21" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>table filtered_feature<span class="sc">-</span>table_sediments.qza \</span>
<span id="cb55-22"><a href="diversity-analyses.html#cb55-22" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>phylogeny <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>asvs<span class="sc">-</span>tree_sediments.qza \</span>
<span id="cb55-23"><a href="diversity-analyses.html#cb55-23" tabindex="-1"></a><span class="sc">--</span>p<span class="sc">-</span>sampling<span class="sc">-</span>depth <span class="dv">20283</span> \</span>
<span id="cb55-24"><a href="diversity-analyses.html#cb55-24" tabindex="-1"></a><span class="sc">--</span>m<span class="sc">-</span>metadata<span class="sc">-</span>file <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>metadata<span class="sc">/</span>metadata<span class="sc">-</span>sediments.csv \</span>
<span id="cb55-25"><a href="diversity-analyses.html#cb55-25" tabindex="-1"></a><span class="sc">--</span>output<span class="sc">-</span>dir <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>diversity<span class="sc">/</span>diversity_sediments</span>
<span id="cb55-26"><a href="diversity-analyses.html#cb55-26" tabindex="-1"></a></span>
<span id="cb55-27"><a href="diversity-analyses.html#cb55-27" tabindex="-1"></a><span class="co"># core metrics for all samples together</span></span>
<span id="cb55-28"><a href="diversity-analyses.html#cb55-28" tabindex="-1"></a>singularity exec <span class="sc">-</span>B <span class="sc">/</span>project <span class="sc">-</span>B <span class="sc">/</span>scratch <span class="sc">/</span>home<span class="sc">/</span><span class="er">&lt;</span><span class="st">&quot;user-ID&quot;</span><span class="sc">&gt;</span><span class="er">/</span>scratch<span class="sc">/</span>qiime2<span class="dv">-2022</span>.<span class="fl">8.</span>sif \</span>
<span id="cb55-29"><a href="diversity-analyses.html#cb55-29" tabindex="-1"></a>qiime diversity core<span class="sc">-</span>metrics<span class="sc">-</span>phylogenetic \</span>
<span id="cb55-30"><a href="diversity-analyses.html#cb55-30" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>table <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>merged<span class="sc">-</span>table.qza \</span>
<span id="cb55-31"><a href="diversity-analyses.html#cb55-31" tabindex="-1"></a><span class="sc">--</span>i<span class="sc">-</span>phylogeny <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>asvs<span class="sc">-</span>tree.qza \</span>
<span id="cb55-32"><a href="diversity-analyses.html#cb55-32" tabindex="-1"></a><span class="sc">--</span>p<span class="sc">-</span>sampling<span class="sc">-</span>depth <span class="dv">20283</span> \</span>
<span id="cb55-33"><a href="diversity-analyses.html#cb55-33" tabindex="-1"></a><span class="sc">--</span>m<span class="sc">-</span>metadata<span class="sc">-</span>file <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>metadata<span class="sc">/</span>metadata.csv \</span>
<span id="cb55-34"><a href="diversity-analyses.html#cb55-34" tabindex="-1"></a><span class="sc">--</span>output<span class="sc">-</span>dir <span class="sc">/</span>data<span class="sc">/</span>amplicon<span class="sc">-</span>analysis<span class="sc">/</span>diversity<span class="sc">/</span>diversity_merged</span></code></pre></div>
<p>Below I’ve provided a brief explanation of each file.</p>
<p>The output artifact (<code>.QZA</code>) files:</p>
<ul>
<li><p><code>rarefied_table.qza</code>: This file contains the rarefied feature table, where the sequencing depth is normalized to ensure an equal number of reads per sample.</p></li>
<li><p><code>faith_pd_vector.qza</code>: This file contains the Faith’s Phylogenetic Diversity values, a measure of diversity that considers the phylogenetic relationships between species.</p></li>
<li><p><code>observed_features_vector.qza</code>: This file contains the number of observed features (e.g., species or OTUs) in each sample.</p></li>
<li><p><code>shannon_vector.qza</code>: This file contains Shannon diversity index values, which measure the diversity of species in each sample, considering both abundance and evenness.</p></li>
<li><p><code>evenness_vector.qza</code>: This file contains evenness values, indicating how evenly the microbial species are distributed within each sample.</p></li>
<li><p><code>unweighted_unifrac_distance_matrix.qza</code>: This file contains the distance matrix computed using the unweighted UniFrac metric, measuring dissimilarity based on the phylogenetic tree.</p></li>
<li><p><code>weighted_unifrac_distance_matrix.qza</code>: Similar to the unweighted UniFrac, this file contains the distance matrix computed using the weighted UniFrac metric.</p></li>
<li><p><code>unweighted_unifrac_pcoa_results.qza</code>: This file contains the Principal Coordinates Analysis (PCoA) results based on the unweighted UniFrac distance metric.</p></li>
<li><p><code>weighted_unifrac_pcoa_results.qza</code>: Similar to the previous one, this file contains PCoA results based on the weighted UniFrac distance metric.</p></li>
</ul>
<blockquote>
<p><strong>NOTE:</strong> Weighted UniFrac in PCA analysis takes into account the abundance of organisms, highlighting the influence of the most abundant species, while unweighted UniFrac focuses on presence and absence, emphasizing less abundant organisms in community comparisons.</p>
</blockquote>
<ul>
<li><p><code>jaccard_distance_matrix.qza</code>: This file contains the distance matrix based on the Jaccard dissimilarity, which measures community dissimilarity based on species presence or absence.</p></li>
<li><p><code>bray_curtis_distance_matrix.qza</code>: This file contains the distance matrix based on the Bray-Curtis dissimilarity, which quantifies compositional differences between samples.</p></li>
<li><p><code>jaccard_pcoa_results.qza</code>: This file contains PCoA results based on the Jaccard dissimilarity.</p></li>
<li><p><code>bray_curtis_pcoa_results.qza</code>: This file contains PCoA results based on the Bray-Curtis dissimilarity.</p></li>
</ul>
<hr />
<p>The output visualization (<code>.QZV</code>) files</p>
<ul>
<li><p><code>unweighted_unifrac_emperor.qzv</code>: This is a visualization file that can be viewed using Emperor, a tool for exploring and visualizing PCoA results.</p></li>
<li><p><code>weighted_unifrac_emperor.qzv</code>: Similar to the previous one, this is a visualization file for PCoA results using the weighted UniFrac metric.</p></li>
<li><p><code>jaccard_emperor.qzv</code>: This is a visualization file for PCoA results based on the Jaccard dissimilarity.</p></li>
<li><p><code>bray_curtis_emperor.qzv</code>: This is a visualization file for PCoA results based on the Bray-Curtis dissimilarity.</p></li>
</ul>
<hr />
 
<hr />
<p style="text-align: center;">
Github: <a href="https://github.com/johannabosch">johannabosch</a> </a>
</p>
<p style="text-align: center;">
<span style="color: #808080;"><em><a href="mailto:yohannabosch@gmail.com" class="email">yohannabosch@gmail.com</a></em></span>
</p>
<!-- Add icon library -->
<p><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"></p>
<!-- Add font awesome icons -->
<p style="text-align: center;">
<a href="https://www.instagram.com/yohannabosch/" class="fa fa-instagram"></a>
<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
<a href="https://github.com/johannabosch/" class="fa fa-github"></a>
</p>
<p> </p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-preprocessing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="assigning-taxonomy.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/07-DiversityAnalyses.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
