<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 VISUALIZING TAXONOMIC DATA | 16S Microbiome Analysis on a Computing Cluster</title>
  <meta name="description" content="9 VISUALIZING TAXONOMIC DATA | 16S Microbiome Analysis on a Computing Cluster" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="9 VISUALIZING TAXONOMIC DATA | 16S Microbiome Analysis on a Computing Cluster" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="johannabosch/QIIME2_for_Graham" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 VISUALIZING TAXONOMIC DATA | 16S Microbiome Analysis on a Computing Cluster" />
  
  
  

<meta name="author" content="Johanna Bosch" />


<meta name="date" content="2023-10-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="assigning-taxonomy.html"/>
<link rel="next" href="resources.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Comparative 16S Amplicon Analysis using QIIME2 on a computing cluster: Tutorial and Resources</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> <strong>INTRODUCTION</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#a-brief-project-review"><i class="fa fa-check"></i><b>1.1</b> <strong>A brief project review</strong></a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introduction.html"><a href="introduction.html#study-species-and-sites"><i class="fa fa-check"></i><b>1.1.1</b> Study species and sites</a></li>
<li class="chapter" data-level="1.1.2" data-path="introduction.html"><a href="introduction.html#sequencing-library-prep"><i class="fa fa-check"></i><b>1.1.2</b> Sequencing &amp; library prep</a></li>
<li class="chapter" data-level="1.1.3" data-path="introduction.html"><a href="introduction.html#files-after-sequencing"><i class="fa fa-check"></i><b>1.1.3</b> Files after sequencing</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#dna-analysis"><i class="fa fa-check"></i><b>1.2</b> <strong>DNA analysis</strong></a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introduction.html"><a href="introduction.html#the-graham-cluster"><i class="fa fa-check"></i><b>1.2.1</b> The Graham cluster</a></li>
<li class="chapter" data-level="1.2.2" data-path="introduction.html"><a href="introduction.html#data-accessibility"><i class="fa fa-check"></i><b>1.2.2</b> Data accessibility</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html"><i class="fa fa-check"></i><b>2</b> <strong>COMPUTE CANADA BASICS</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#cluster-computing-basics"><i class="fa fa-check"></i><b>2.1</b> <strong>Cluster computing basics</strong></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#signing-into-the-cluster"><i class="fa fa-check"></i><b>2.1.1</b> Signing into the cluster</a></li>
<li class="chapter" data-level="2.1.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#basic-computing-commands"><i class="fa fa-check"></i><b>2.1.2</b> Basic computing commands</a></li>
<li class="chapter" data-level="2.1.3" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#loading-modules"><i class="fa fa-check"></i><b>2.1.3</b> Loading modules</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#job-scripts"><i class="fa fa-check"></i><b>2.2</b> <strong>Job scripts </strong></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#using-nano"><i class="fa fa-check"></i><b>2.2.1</b> Using Nano</a></li>
<li class="chapter" data-level="2.2.2" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#writing-a-bash-script"><i class="fa fa-check"></i><b>2.2.2</b> Writing a bash script</a></li>
<li class="chapter" data-level="2.2.3" data-path="compute-canada-basics.html"><a href="compute-canada-basics.html#running-scripts-on-the-scheduler"><i class="fa fa-check"></i><b>2.2.3</b> Running scripts on the scheduler</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="importing-data.html"><a href="importing-data.html"><i class="fa fa-check"></i><b>3</b> <strong>IMPORTING DATA</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="importing-data.html"><a href="importing-data.html#getting-organized"><i class="fa fa-check"></i><b>3.1</b> <strong>Getting organized</strong></a></li>
<li class="chapter" data-level="3.2" data-path="importing-data.html"><a href="importing-data.html#moving-data-to-the-cluster"><i class="fa fa-check"></i><b>3.2</b> <strong>Moving data to the cluster</strong></a></li>
<li class="chapter" data-level="3.3" data-path="importing-data.html"><a href="importing-data.html#organizing-.fastq-files"><i class="fa fa-check"></i><b>3.3</b> <strong>Organizing .FASTQ files</strong></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="importing-data.html"><a href="importing-data.html#unzip-rawdata-files"><i class="fa fa-check"></i><b>3.3.1</b> Unzip rawdata files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html"><i class="fa fa-check"></i><b>4</b> <strong>QIIME2 ON THE CLUSTER</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#use-qiime2-via-apptainer"><i class="fa fa-check"></i><b>4.1</b> <strong>Use QIIME2 via Apptainer</strong></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#finding-physical-paths"><i class="fa fa-check"></i><b>4.1.1</b> Finding physical paths</a></li>
<li class="chapter" data-level="4.1.2" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#using-apptainer-mounts"><i class="fa fa-check"></i><b>4.1.2</b> Using Apptainer mounts</a></li>
<li class="chapter" data-level="4.1.3" data-path="qiime2-on-the-cluster.html"><a href="qiime2-on-the-cluster.html#getting-help"><i class="fa fa-check"></i><b>4.1.3</b> Getting help</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html"><i class="fa fa-check"></i><b>5</b> <strong>GETTING STARTED WITH QIIME2</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#creating-artifact-files"><i class="fa fa-check"></i><b>5.1</b> <strong>Creating artifact files</strong></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#import-the-.fastq-files"><i class="fa fa-check"></i><b>5.1.1</b> Import the .FASTQ files</a></li>
<li class="chapter" data-level="5.1.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#inspect-the-.qza-files"><i class="fa fa-check"></i><b>5.1.2</b> Inspect the .QZA files</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#visualizing-artifact-files"><i class="fa fa-check"></i><b>5.2</b> <strong>Visualizing artifact files</strong></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#create-a-summary-report"><i class="fa fa-check"></i><b>5.2.1</b> Create a summary report</a></li>
<li class="chapter" data-level="5.2.2" data-path="getting-started-with-qiime2.html"><a href="getting-started-with-qiime2.html#using-qiime2s-web-app"><i class="fa fa-check"></i><b>5.2.2</b> Using QIIME2’s web-app</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-preprocessing.html"><a href="data-preprocessing.html"><i class="fa fa-check"></i><b>6</b> <strong>DATA PREPROCESSING</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#quality-assessment"><i class="fa fa-check"></i><b>6.1</b> <strong>Quality assessment</strong></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#fastqc"><i class="fa fa-check"></i><b>6.1.1</b> FastQC</a></li>
<li class="chapter" data-level="6.1.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#multiqc"><i class="fa fa-check"></i><b>6.1.2</b> MultiQC</a></li>
<li class="chapter" data-level="6.1.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#visualizing-data-quality"><i class="fa fa-check"></i><b>6.1.3</b> Visualizing data quality</a></li>
<li class="chapter" data-level="6.1.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#plot-types"><i class="fa fa-check"></i><b>6.1.4</b> Plot types</a></li>
<li class="chapter" data-level="6.1.5" data-path="data-preprocessing.html"><a href="data-preprocessing.html#for-this-study"><i class="fa fa-check"></i><b>6.1.5</b> For this study</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-cleaning"><i class="fa fa-check"></i><b>6.2</b> <strong>Data Cleaning</strong></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#trimming-primers"><i class="fa fa-check"></i><b>6.2.1</b> Trimming primers</a></li>
<li class="chapter" data-level="6.2.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#summarizing-data"><i class="fa fa-check"></i><b>6.2.2</b> Summarizing data</a></li>
<li class="chapter" data-level="6.2.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#joining-read-pairs"><i class="fa fa-check"></i><b>6.2.3</b> Joining read pairs</a></li>
<li class="chapter" data-level="6.2.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#quality-filtering"><i class="fa fa-check"></i><b>6.2.4</b> Quality filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-transformation"><i class="fa fa-check"></i><b>6.3</b> <strong>Data transformation</strong></a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#constructing-asvs"><i class="fa fa-check"></i><b>6.3.1</b> Constructing ASV’s</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#data-reduction"><i class="fa fa-check"></i><b>6.4</b> <strong>Data reduction</strong></a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#filtering-unique-reads"><i class="fa fa-check"></i><b>6.4.1</b> Filtering unique reads</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="data-preprocessing.html"><a href="data-preprocessing.html#manipulating-tables"><i class="fa fa-check"></i><b>6.5</b> <strong>Manipulating tables</strong></a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#merging-feature-tables"><i class="fa fa-check"></i><b>6.5.1</b> Merging feature tables</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="diversity-analyses.html"><a href="diversity-analyses.html"><i class="fa fa-check"></i><b>7</b> <strong>DIVERSITY ANALYSES</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#build-a-phylogenetic-tree"><i class="fa fa-check"></i><b>7.1</b> <strong>Build a phylogenetic tree</strong></a></li>
<li class="chapter" data-level="7.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#generating-alpha-rarefaction-curves"><i class="fa fa-check"></i><b>7.2</b> <strong>Generating alpha rarefaction curves</strong></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#finding-max-depth"><i class="fa fa-check"></i><b>7.2.1</b> Finding max depth</a></li>
<li class="chapter" data-level="7.2.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#generating-the-curve"><i class="fa fa-check"></i><b>7.2.2</b> Generating the curve</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-and-beta-diversity"><i class="fa fa-check"></i><b>7.3</b> <strong>Alpha and beta diversity</strong></a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#running-diversity-analyses"><i class="fa fa-check"></i><b>7.3.1</b> Running diversity analyses</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html"><i class="fa fa-check"></i><b>8</b> <strong>ASSIGNING TAXONOMY</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#classifying-asvs"><i class="fa fa-check"></i><b>8.1</b> <strong>Classifying ASV’s</strong></a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#using-scikit-learn"><i class="fa fa-check"></i><b>8.1.1</b> Using scikit-learn</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#verifying-with-blastn"><i class="fa fa-check"></i><b>8.2</b> <strong>Verifying with BLASTn</strong></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#filtering-contaminants"><i class="fa fa-check"></i><b>8.2.1</b> Filtering contaminants</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="assigning-taxonomy.html"><a href="assigning-taxonomy.html#grouping-feature-tables"><i class="fa fa-check"></i><b>8.3</b> <strong>Grouping feature tables</strong></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html"><i class="fa fa-check"></i><b>9</b> <strong>VISUALIZING TAXONOMIC DATA</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#exporting-data"><i class="fa fa-check"></i><b>9.1</b> <strong>Exporting data</strong></a></li>
<li class="chapter" data-level="9.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#importing-q2-files-to-r"><i class="fa fa-check"></i><b>9.2</b> <strong>Importing Q2 files to R</strong></a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#qiime2r"><i class="fa fa-check"></i><b>9.3</b> <strong>QIIME2R</strong></a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-feature-table"><i class="fa fa-check"></i><b>9.3.1</b> Load the feature table</a></li>
<li class="chapter" data-level="9.3.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-metadata"><i class="fa fa-check"></i><b>9.3.2</b> Load the metadata</a></li>
<li class="chapter" data-level="9.3.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#load-the-taxonomic-data"><i class="fa fa-check"></i><b>9.3.3</b> Load the taxonomic data</a></li>
<li class="chapter" data-level="9.3.4" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#parsing-taxonomic-data"><i class="fa fa-check"></i><b>9.3.4</b> Parsing taxonomic data</a></li>
<li class="chapter" data-level="9.3.5" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#create-a-taxa-barplot"><i class="fa fa-check"></i><b>9.3.5</b> Create a taxa barplot</a></li>
<li class="chapter" data-level="9.3.6" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#create-a-taxa-heatmap"><i class="fa fa-check"></i><b>9.3.6</b> Create a taxa heatmap</a></li>
<li class="chapter" data-level="9.3.7" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#plot-beta-diversity"><i class="fa fa-check"></i><b>9.3.7</b> Plot beta-diversity</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#phyloseq"><i class="fa fa-check"></i><b>9.4</b> <strong>Phyloseq</strong></a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#build-a-phyloseq-object"><i class="fa fa-check"></i><b>9.4.1</b> Build a phyloseq object</a></li>
<li class="chapter" data-level="9.4.2" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#taxonomic-characteristics"><i class="fa fa-check"></i><b>9.4.2</b> Taxonomic characteristics</a></li>
<li class="chapter" data-level="9.4.3" data-path="visualizing-taxonomic-data.html"><a href="visualizing-taxonomic-data.html#plot-alpha-diversity"><i class="fa fa-check"></i><b>9.4.3</b> Plot alpha-diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>10</b> <strong>Resources:</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="resources.html"><a href="resources.html#data-in-this-analysis"><i class="fa fa-check"></i><b>10.1</b> <strong>Data in this analysis</strong></a></li>
<li class="chapter" data-level="10.2" data-path="resources.html"><a href="resources.html#other-resources"><i class="fa fa-check"></i><b>10.2</b> <strong>Other resources</strong></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i><b>11</b> Acknowledgments</a></li>
<li class="chapter" data-level="12" data-path="technical-support.html"><a href="technical-support.html"><i class="fa fa-check"></i><b>12</b> <strong>Technical Support</strong></a></li>
<li class="chapter" data-level="13" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>13</b> <strong>References</strong></a>
<ul>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html#references-1"><i class="fa fa-check"></i><strong>References</strong></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">16S Microbiome Analysis on a Computing Cluster</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visualizing-taxonomic-data" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">9</span> <strong>VISUALIZING TAXONOMIC DATA</strong><a href="visualizing-taxonomic-data.html#visualizing-taxonomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>You can build lots of different charts through QIIME2 via the compute canada clusters, like we did with the diversity plots - but I think the QIIME2 plots are not as versatile as the ones that can be built in R. Here, I walk you through the steps of exporting the appropriate data to R to use several different packages that handle QIIME2 artifacts. The figures made in these next visualization sections can be saved as a vector file and then edited in other software such as Illustrator, Power Point, Libre Office, etc. for publishing.</p>
<p>If you’re brand new to using R and RStudio, you should become familiar with the interface first. Here are a few links to get started</p>
<p><a href="https://education.rstudio.com/learn/beginner/">REducation Beginners Tutorial</a></p>
<p><a href="https://ahurford.github.io/quant-guide-all-courses/install.html">Quantitative Guide for Biology: Intro to R</a></p>
<div id="exporting-data" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> <strong>Exporting data</strong><a href="visualizing-taxonomic-data.html#exporting-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To begin making the figures, let’s export the data on to a local computer from the cluster using the safe file transfer protocol (SFTP). You can return to <strong>Chapter 03 - Importing Data</strong> to review how the SFTP works.</p>
<p>I exported the following QZA files onto my local computer and kept them in a file called <code>Desktop/QIIME_files</code>. <strong>Make sure to put these files into a dedicated folder for the remainder of the analysis so they are easy to find</strong></p>
<table>
<colgroup>
<col width="24%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>File type</th>
<th>Associated file names</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Final feature table</td>
<td>final_table-sediments.qza, final_table-birds.qza, final_merged-table.qza</td>
</tr>
<tr class="even">
<td>Representative sequences</td>
<td>rep-seqs_sediments.qza, rep-seqs_birds.qza, rep_seqs.qza</td>
</tr>
<tr class="odd">
<td>Classification files</td>
<td>classification_sediments.qza, classification_birds.qza, classification.qza</td>
</tr>
<tr class="even">
<td>Phylogenetic tree</td>
<td>asvs-tree_sediments.qza, insertion-placements_sediments.qza, asvs-tree_birds.qza, insertion-placements_birds.qza, asvs-tree.qza, insertion-placements.qza</td>
</tr>
<tr class="odd">
<td>Merged feature table</td>
<td>merged-table.qza</td>
</tr>
<tr class="even">
<td>Grouped feature table</td>
<td>grouped-table.qza</td>
</tr>
<tr class="odd">
<td>Diversity folder for birds</td>
<td>includes all of the files from the <code>qiime core-metrics-phylogenetic</code> command run on the bird samples (put it into a folder named <code>/diversity-birds</code>)</td>
</tr>
<tr class="even">
<td>Diversity folder for sediments</td>
<td>includes all of the files from the <code>qiime core-metrics-phylogenetic</code> command run on the sediment samples (put it into a folder named <code>/diversity-sediments</code>)</td>
</tr>
<tr class="odd">
<td>Diversity folder for merged files</td>
<td>includes all of the files from the <code>qiime core-metrics-phylogenetic</code> command run on the merged feature table (put it into a folder named <code>/diversity-merged</code>)</td>
</tr>
</tbody>
</table>
</div>
<div id="importing-q2-files-to-r" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> <strong>Importing Q2 files to R</strong><a href="visualizing-taxonomic-data.html#importing-q2-files-to-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A few different packages are used to load the QIIME2 artifacts into R. If you’re new to R and haven’t installed Rtools, do that first. RTools is a separate software that you have to download and install in your system (not in R). Follow this link to install RTools on your system - <a href="https://cran.rstudio.com/bin/windows/Rtools/" class="uri">https://cran.rstudio.com/bin/windows/Rtools/</a>. Sometimes you can get away without doing this step, it doesn’t hurt to try!</p>
<p>Begin by installing and loading the following R packages. To install the qiime2R package, I used the remotes package because some versions of R will not allow you to install qiime2R through the normal <code>install.packages</code> method.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="visualizing-taxonomic-data.html#cb61-1" tabindex="-1"></a><span class="co">#install the ggforce and remotes package</span></span>
<span id="cb61-2"><a href="visualizing-taxonomic-data.html#cb61-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;remotes&quot;</span>)</span>
<span id="cb61-3"><a href="visualizing-taxonomic-data.html#cb61-3" tabindex="-1"></a></span>
<span id="cb61-4"><a href="visualizing-taxonomic-data.html#cb61-4" tabindex="-1"></a><span class="co">#now load the remotes package</span></span>
<span id="cb61-5"><a href="visualizing-taxonomic-data.html#cb61-5" tabindex="-1"></a><span class="fu">library</span>(remotes)</span>
<span id="cb61-6"><a href="visualizing-taxonomic-data.html#cb61-6" tabindex="-1"></a></span>
<span id="cb61-7"><a href="visualizing-taxonomic-data.html#cb61-7" tabindex="-1"></a><span class="co">#install qiime2R</span></span>
<span id="cb61-8"><a href="visualizing-taxonomic-data.html#cb61-8" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jbisanz/qiime2R&quot;</span>)</span></code></pre></div>
<p>Now install and load some other packages that will be used in the downstream analysis</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="visualizing-taxonomic-data.html#cb62-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb62-2"><a href="visualizing-taxonomic-data.html#cb62-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggforce&quot;</span>)</span>
<span id="cb62-3"><a href="visualizing-taxonomic-data.html#cb62-3" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="visualizing-taxonomic-data.html#cb63-1" tabindex="-1"></a><span class="co">#load the packages</span></span>
<span id="cb63-2"><a href="visualizing-taxonomic-data.html#cb63-2" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb63-3"><a href="visualizing-taxonomic-data.html#cb63-3" tabindex="-1"></a><span class="fu">library</span>(ggforce)</span>
<span id="cb63-4"><a href="visualizing-taxonomic-data.html#cb63-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div id="qiime2r" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> <strong>QIIME2R</strong><a href="visualizing-taxonomic-data.html#qiime2r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://github.com/jbisanz/qiime2R">QIIME2R</a> is an R package that was created to manipulate QIIME2 artifact files for downstream analysis. It allows users to extract data from an artifact file, manipulate the data within the files, and create publication quality figures, including taxonomic bar charts, diversity plots, and more.</p>
<p>Start by setting your working directory, here I use the file <code>Desktop/QIIME-files</code></p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="visualizing-taxonomic-data.html#cb64-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;C:/Users/Hanna/Desktop/QIIME_files&quot;</span>)</span>
<span id="cb64-2"><a href="visualizing-taxonomic-data.html#cb64-2" tabindex="-1"></a></span>
<span id="cb64-3"><a href="visualizing-taxonomic-data.html#cb64-3" tabindex="-1"></a><span class="co">#check what files are in your wd</span></span>
<span id="cb64-4"><a href="visualizing-taxonomic-data.html#cb64-4" tabindex="-1"></a><span class="fu">list.files</span>()</span></code></pre></div>
<p>To begin using QIIME2R, make sure the package is loaded first.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="visualizing-taxonomic-data.html#cb65-1" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span></code></pre></div>
<div id="load-the-feature-table" class="section level3 hasAnchor" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Load the feature table<a href="visualizing-taxonomic-data.html#load-the-feature-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, read the feature tables in .QZA format that we will be using for the downstream analysis. Here I use the <code>&lt;-</code> to assign the term <code>bird_features</code> to the file containing the feature counts for every bird sample (found in the folder QIIME-files in my Desktop folder).</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="visualizing-taxonomic-data.html#cb66-1" tabindex="-1"></a>bird_features<span class="ot">&lt;-</span> <span class="fu">read_qza</span>(<span class="st">&quot;final_table-birds.qza&quot;</span>)</span></code></pre></div>
<p>Once the data is read, you can view it using this command:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="visualizing-taxonomic-data.html#cb67-1" tabindex="-1"></a><span class="fu">view</span>(bird_features<span class="sc">$</span>data)</span></code></pre></div>
<p>Now do the same for the sediment samples and the grouped feature table that was made in Ch-06 (Data Preprocessing):</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="visualizing-taxonomic-data.html#cb68-1" tabindex="-1"></a>sed_features <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(<span class="st">&quot;final_table-sediments.qza&quot;</span>)</span>
<span id="cb68-2"><a href="visualizing-taxonomic-data.html#cb68-2" tabindex="-1"></a><span class="fu">view</span>(sed_features<span class="sc">$</span>data)</span>
<span id="cb68-3"><a href="visualizing-taxonomic-data.html#cb68-3" tabindex="-1"></a></span>
<span id="cb68-4"><a href="visualizing-taxonomic-data.html#cb68-4" tabindex="-1"></a>grouped_features <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(<span class="st">&quot;grouped-table.qza&quot;</span>)</span>
<span id="cb68-5"><a href="visualizing-taxonomic-data.html#cb68-5" tabindex="-1"></a><span class="fu">view</span>(grouped_features<span class="sc">$</span>data)</span></code></pre></div>
<p>To view if the information stored in the files is correct you can also use the<code>names</code> command:</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="visualizing-taxonomic-data.html#cb69-1" tabindex="-1"></a><span class="fu">names</span>(sed_features) <span class="co">#shows first 5 samples and first 5 taxa</span></span>
<span id="cb69-2"><a href="visualizing-taxonomic-data.html#cb69-2" tabindex="-1"></a><span class="fu">names</span>(bird_features)</span></code></pre></div>
<p>One handy command to know is the <code>write.csv()</code> command, which can be used to create .csv file of your feature table.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="visualizing-taxonomic-data.html#cb70-1" tabindex="-1"></a><span class="fu">write.csv</span>(bird_features<span class="sc">$</span>data, <span class="st">&quot;feature-table_birds.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="load-the-metadata" class="section level3 hasAnchor" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Load the metadata<a href="visualizing-taxonomic-data.html#load-the-metadata" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that the feature table is loaded into R, the metadata can also be loaded. Always ensure that your metadata file is properly formatted.</p>
<p>If it’s unclear how metadata is used in this analysis, you should download the <a href="https://github.com/johannabosch/QIIME2_for_Graham/blob/main">metadata files</a> titled <code>metadata.csv</code>, <code>birds_metadata.csv</code>, and <code>sediments_metadata.csv</code>. Take a look at the formatting of each file, and the type of data that’s attached to each sample.</p>
<p>Metadata in microbiome analysis is akin to the footnotes in a research paper; while not essential for understanding the main text, it provides valuable information about the samples, such as site ID, sampling location, and species Latin names. This additional context enhances the comprehensibility and depth of taxonomic bar charts, diversity box plots, and other figures. The first column, labelled <code>#SampleID</code> is a special column header indicating the sample IDs, the other categories are the metadata. Each row represents a sample, with corresponding values for each metadata category. Remember that the specific metadata categories and their names can vary depending on your study and the information you want to associate with your samples. To learn more about using metadata in QIIME2, visit the <a href="https://docs.qiime2.org/2023.7/tutorials/metadata/#metadata-columns">Metadata in QIIME2</a> page.</p>
<p>Load the metadata into R:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="visualizing-taxonomic-data.html#cb71-1" tabindex="-1"></a><span class="co">#metadata file for only sediments</span></span>
<span id="cb71-2"><a href="visualizing-taxonomic-data.html#cb71-2" tabindex="-1"></a>sediment_metadata<span class="ot">&lt;-</span><span class="fu">read_q2metadata</span>(<span class="st">&quot;metadata-sediments.csv&quot;</span>)</span>
<span id="cb71-3"><a href="visualizing-taxonomic-data.html#cb71-3" tabindex="-1"></a></span>
<span id="cb71-4"><a href="visualizing-taxonomic-data.html#cb71-4" tabindex="-1"></a><span class="co">#metadata file for only bird fecal samples</span></span>
<span id="cb71-5"><a href="visualizing-taxonomic-data.html#cb71-5" tabindex="-1"></a>bird_metadata<span class="ot">&lt;-</span><span class="fu">read_q2metadata</span>(<span class="st">&quot;metadata-birds.csv&quot;</span>)</span>
<span id="cb71-6"><a href="visualizing-taxonomic-data.html#cb71-6" tabindex="-1"></a></span>
<span id="cb71-7"><a href="visualizing-taxonomic-data.html#cb71-7" tabindex="-1"></a><span class="co">#metadata file containing all the samples</span></span>
<span id="cb71-8"><a href="visualizing-taxonomic-data.html#cb71-8" tabindex="-1"></a>metadata<span class="ot">&lt;-</span><span class="fu">read_q2metadata</span>(<span class="st">&quot;metadata.csv&quot;</span>)</span>
<span id="cb71-9"><a href="visualizing-taxonomic-data.html#cb71-9" tabindex="-1"></a></span>
<span id="cb71-10"><a href="visualizing-taxonomic-data.html#cb71-10" tabindex="-1"></a><span class="co">#metadata file containing all the grouped samples</span></span>
<span id="cb71-11"><a href="visualizing-taxonomic-data.html#cb71-11" tabindex="-1"></a><span class="co">#you do not have to use the QIIME2R&#39;s command, </span></span>
<span id="cb71-12"><a href="visualizing-taxonomic-data.html#cb71-12" tabindex="-1"></a>grouped_metadata<span class="ot">&lt;-</span><span class="fu">read_q2metadata</span>(<span class="st">&quot;metadata-grouped.csv&quot;</span>)</span>
<span id="cb71-13"><a href="visualizing-taxonomic-data.html#cb71-13" tabindex="-1"></a></span>
<span id="cb71-14"><a href="visualizing-taxonomic-data.html#cb71-14" tabindex="-1"></a><span class="fu">head</span>(metadata) <span class="co">#used to show top lines of a metadata file</span></span></code></pre></div>
</div>
<div id="load-the-taxonomic-data" class="section level3 hasAnchor" number="9.3.3">
<h3><span class="header-section-number">9.3.3</span> Load the taxonomic data<a href="visualizing-taxonomic-data.html#load-the-taxonomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The next step to visualizing data using QIIME2R is to load the taxonomy data into R. This is done using the same <code>read_qza</code> command that was used before.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="visualizing-taxonomic-data.html#cb72-1" tabindex="-1"></a>taxonomy<span class="ot">&lt;-</span><span class="fu">read_qza</span>(<span class="st">&quot;classification.qza&quot;</span>)</span>
<span id="cb72-2"><a href="visualizing-taxonomic-data.html#cb72-2" tabindex="-1"></a></span>
<span id="cb72-3"><a href="visualizing-taxonomic-data.html#cb72-3" tabindex="-1"></a><span class="fu">head</span>(taxonomy<span class="sc">$</span>data) <span class="co">#used to see the first lines of the taxonomy data file</span></span>
<span id="cb72-4"><a href="visualizing-taxonomic-data.html#cb72-4" tabindex="-1"></a></span>
<span id="cb72-5"><a href="visualizing-taxonomic-data.html#cb72-5" tabindex="-1"></a><span class="fu">view</span>(taxonomy<span class="sc">$</span>data) <span class="co">#used to open to taxonomy file in a new tab</span></span></code></pre></div>
</div>
<div id="parsing-taxonomic-data" class="section level3 hasAnchor" number="9.3.4">
<h3><span class="header-section-number">9.3.4</span> Parsing taxonomic data<a href="visualizing-taxonomic-data.html#parsing-taxonomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>An important step that should be considered is parsing the <code>classification.qza</code> file to break up the taxonomic levels. This will separate each level of classification given to a feature. Here, the term <code>taxonomy</code> is being assigned to the part of the <code>classification.qza</code> file that contains the data (<code>taxonomy$data</code>). In other words, the term <code>taxonomy</code> now becomes the name of the variable where the parsed taxonomic data will be stored. <code>parse_taxonomy</code> is simply a function in the QIIME2R package specifically designed to extract taxonomic data from QIIME 2 artifacts. For example, a string of taxonomic classifications that looks like the first table shown here will end up looking like the second table:</p>
<p><strong>Table 1. Original taxonomic file:</strong></p>
<table>
<colgroup>
<col width="42%" />
<col width="57%" />
</colgroup>
<thead>
<tr class="header">
<th>FeatureID</th>
<th>Taxonomy</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>4595dec6ed…</td>
<td>k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Lachnospiraceae;g__Roseburia;s__Roseburia intestinalis</td>
</tr>
</tbody>
</table>
<p><strong>Table 2. Parsed taxonomic data file:</strong></p>
<table>
<colgroup>
<col width="12%" />
<col width="14%" />
<col width="12%" />
<col width="14%" />
<col width="12%" />
<col width="12%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th>FeatureID</th>
<th>Kingdom</th>
<th>Phylum</th>
<th>Class</th>
<th>Order</th>
<th>Family</th>
<th>Genus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>4595dec6ed…</td>
<td>Bacteria</td>
<td>Firmicutes</td>
<td>Clostridia</td>
<td>Clostridiales</td>
<td>Lachnospiraceae</td>
<td>Roseburia</td>
</tr>
</tbody>
</table>
<p>To parse the taxonomy file, use the following command:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="visualizing-taxonomic-data.html#cb73-1" tabindex="-1"></a>taxonomy<span class="ot">&lt;-</span><span class="fu">parse_taxonomy</span>(taxonomy<span class="sc">$</span>data)</span>
<span id="cb73-2"><a href="visualizing-taxonomic-data.html#cb73-2" tabindex="-1"></a><span class="fu">view</span>(taxonomy)</span>
<span id="cb73-3"><a href="visualizing-taxonomic-data.html#cb73-3" tabindex="-1"></a></span>
<span id="cb73-4"><a href="visualizing-taxonomic-data.html#cb73-4" tabindex="-1"></a><span class="co">#save a .csv formatted version of the file if you want</span></span>
<span id="cb73-5"><a href="visualizing-taxonomic-data.html#cb73-5" tabindex="-1"></a><span class="fu">write.csv</span>(taxonomy, <span class="st">&quot;taxa.csv&quot;</span>) </span></code></pre></div>
<p><strong>NOTE:</strong> when you do this, you will replace your original file assigned to the term <code>taxonomy</code> with the new parsed file that was also assigned to the term <code>taxonomy</code>. If you want to keep both files separate, you will have to change the term assigned to the parsed file to something like <code>parsed-taxonomy</code>.</p>
</div>
<div id="create-a-taxa-barplot" class="section level3 hasAnchor" number="9.3.5">
<h3><span class="header-section-number">9.3.5</span> Create a taxa barplot<a href="visualizing-taxonomic-data.html#create-a-taxa-barplot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this next section, I create a taxonomic bar chart which presents the top 10 most abundant phyla across all of my seabird and sediment samples together. This means that the resulting plot will show the top 10 most abundant phyla for every seabird species or sediment sampling site side by side. To do this we have to use the <code>grouped-table.qza</code> file that contains feature counts for both the sediment sampling sites and seabird species that was made in the last steps of Ch-06 (Data Preprocessing).</p>
<p>Make sure the following packages are loaded for this next section</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="visualizing-taxonomic-data.html#cb74-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb74-2"><a href="visualizing-taxonomic-data.html#cb74-2" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb74-3"><a href="visualizing-taxonomic-data.html#cb74-3" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span></code></pre></div>
<p>Next, load the colors you’d like to use for your bar plot. This can be done in many different ways. You can always use a pre-made color palette that can be loaded through various R packages.</p>
<p>Let’s try that first! Start by trying to load the RColorBrewer package and choosing the <code>Spectral</code> palette. Here I specify how many colors are needed for each sample group with <code>n=10</code>.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="visualizing-taxonomic-data.html#cb75-1" tabindex="-1"></a><span class="co"># method 1: loading a color palette </span></span>
<span id="cb75-2"><a href="visualizing-taxonomic-data.html#cb75-2" tabindex="-1"></a></span>
<span id="cb75-3"><a href="visualizing-taxonomic-data.html#cb75-3" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;RColorBrewer&quot;</span>)</span>
<span id="cb75-4"><a href="visualizing-taxonomic-data.html#cb75-4" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb75-5"><a href="visualizing-taxonomic-data.html#cb75-5" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="at">name=</span><span class="st">&quot;Spectral&quot;</span>, <span class="at">n =</span> <span class="dv">10</span>), <span class="fu">brewer.pal</span>(<span class="at">name=</span><span class="st">&quot;BrBG&quot;</span>, <span class="at">n =</span> <span class="dv">10</span>))</span></code></pre></div>
<p>If you have more sampling groups than the color palette has colors, then this method will not work. You can also combine color palettes, but at that point I just prefer to specifiy the colors myself. And in many cases, assigning your own unique HEX codes to each group is really useful. In cases where you’re working with a lot of different samples, you’ll need the colors to be distinguishable from one another when sitting side by side in the barplot. By assigning your own colors to groups you can always go back and switch the order of colors in your plot.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="visualizing-taxonomic-data.html#cb76-1" tabindex="-1"></a><span class="co">#method 2: assigning your own colors</span></span>
<span id="cb76-2"><a href="visualizing-taxonomic-data.html#cb76-2" tabindex="-1"></a></span>
<span id="cb76-3"><a href="visualizing-taxonomic-data.html#cb76-3" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#FF0000&quot;</span>, <span class="st">&quot;#FFA500&quot;</span>, <span class="st">&quot;#FFFF00&quot;</span>, <span class="st">&quot;#00FF00&quot;</span>, <span class="st">&quot;#00FFFF&quot;</span>, <span class="st">&quot;#0000FF&quot;</span>, <span class="st">&quot;#800080&quot;</span>, <span class="st">&quot;#FF00FF&quot;</span>, <span class="st">&quot;#FF4500&quot;</span>, <span class="st">&quot;#FF69B4&quot;</span>, <span class="st">&quot;#004445&quot;</span>)</span></code></pre></div>
<p>This next line of code specifies the dimensions of your plot, which is sometimes necessary for plots that are really large but may not necessary.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="visualizing-taxonomic-data.html#cb77-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">1000</span>,<span class="dv">1000</span>,<span class="dv">1000</span>,<span class="dv">1000</span>))</span></code></pre></div>
<p>Make sure the grouped-table.qza file (assigned to the term <code>grouped_features</code>) and the metadata.csv file (assigned to the term <code>metadata</code>) is loaded into R. Next, use the <code>summarize_taxa</code> command to summarize the taxonomy file by phylum. This line of code can also be used to summarize data by genera, or any other taxonomic level present in your parsed taxonomy file, simply change the term after the <code>$</code> in this code.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="visualizing-taxonomic-data.html#cb78-1" tabindex="-1"></a>taxasums <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(grouped_features<span class="sc">$</span>data, taxonomy)<span class="sc">$</span>Phylum</span></code></pre></div>
<p>Open this new <code>taxasums</code> file to see what happened here. This command merged the taxonomy file and the grouped feature table, and replaced the feature ID’s with their associated taxa, meanwhile keeping the feature counts present for each group.</p>
<p>Now use this command to plot the bar graph. Here, I specify that the data should be plotted according to the <code>group</code> column in the file <code>metadata.csv</code>, and that 20 phyla should be plotted to the graph.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="visualizing-taxonomic-data.html#cb79-1" tabindex="-1"></a><span class="fu">taxa_barplot</span>(taxasums, grouped_metadata, <span class="st">&quot;group&quot;</span>, <span class="at">ntoplot=</span><span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb79-2"><a href="visualizing-taxonomic-data.html#cb79-2" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> colors) <span class="sc">+</span> <span class="co">#add colors we chose to the plot</span></span>
<span id="cb79-3"><a href="visualizing-taxonomic-data.html#cb79-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="sc">+</span> <span class="co">#add a theme to change font size</span></span>
<span id="cb79-4"><a href="visualizing-taxonomic-data.html#cb79-4" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="cn">FALSE</span>) <span class="co">#remove this line if you want a legend</span></span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:fig1"></span>
<img src="images/taxa_barplot.png" alt="Raw version of the taxa barplot showing the top 10 most abundant phyla present in each sampling group"  />
<p class="caption">
Figure 9.1: Raw version of the taxa barplot showing the top 10 most abundant phyla present in each sampling group
</p>
</div>
<p>Save the figure as an .SVG file so that you can edit it in another software by ungrouping the single layer file. I like using LibreOffice Draw on Linux, or you can use PowerPoint, Illustrator, etc. If you don’t know how to do this in your chosen software, Google it!</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="visualizing-taxonomic-data.html#cb80-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;taxa_birds.svg&quot;</span>, <span class="at">height =</span> <span class="dv">50</span>, <span class="at">width =</span> <span class="dv">100</span>, <span class="at">units=</span><span class="st">&quot;cm&quot;</span>)</span></code></pre></div>
<p>If you’d like to know the exact feature counts used in this graph, create a .csv file here that contains all of the feature counts from the taxasums file</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="visualizing-taxonomic-data.html#cb81-1" tabindex="-1"></a><span class="fu">write.csv</span>(taxasums, <span class="st">&quot;top-taxa-phyla.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="create-a-taxa-heatmap" class="section level3 hasAnchor" number="9.3.6">
<h3><span class="header-section-number">9.3.6</span> Create a taxa heatmap<a href="visualizing-taxonomic-data.html#create-a-taxa-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now let’s make two heatmaps that show the top 50 most abundant genera in (1) the sediment samples and (2) the seabird fecal samples. Taxonomic heatmaps that highlight the most abundant genera in your data set are important in microbiome studies because they provide a quick and visually intuitive way to identify the key microbial taxa driving differences between samples or experimental groups. These heat maps help researchers pinpoint which genera are significantly contributing to variations in microbial composition, aiding in the discovery of potential biomarkers or ecological patterns. Additionally, they simplify the complex data by focusing on the most relevant taxa, making it easier to interpret and communicate findings to a broader audience.</p>
<p>Here’s a review on how to set up the data for the taxonomic heatmaps. If you’re having trouble understanding some of these concepts, revisit the previous section on taxonomic bar plots for a more thorough review on the commands.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="visualizing-taxonomic-data.html#cb82-1" tabindex="-1"></a><span class="co">#summarize the taxa based on genera for each sample type</span></span>
<span id="cb82-2"><a href="visualizing-taxonomic-data.html#cb82-2" tabindex="-1"></a>taxasums_birds <span class="ot">&lt;-</span><span class="fu">summarize_taxa</span>(bird_features<span class="sc">$</span>data, taxonomy)<span class="sc">$</span>Genus</span>
<span id="cb82-3"><a href="visualizing-taxonomic-data.html#cb82-3" tabindex="-1"></a>taxasums_sediments <span class="ot">&lt;-</span><span class="fu">summarize_taxa</span>(sed_features<span class="sc">$</span>data, taxonomy)<span class="sc">$</span>Genus</span>
<span id="cb82-4"><a href="visualizing-taxonomic-data.html#cb82-4" tabindex="-1"></a></span>
<span id="cb82-5"><a href="visualizing-taxonomic-data.html#cb82-5" tabindex="-1"></a><span class="co">#check the new taxasums files </span></span>
<span id="cb82-6"><a href="visualizing-taxonomic-data.html#cb82-6" tabindex="-1"></a><span class="fu">view</span>(taxasums_birds)</span>
<span id="cb82-7"><a href="visualizing-taxonomic-data.html#cb82-7" tabindex="-1"></a><span class="fu">view</span>(taxasums_sediments)</span>
<span id="cb82-8"><a href="visualizing-taxonomic-data.html#cb82-8" tabindex="-1"></a></span>
<span id="cb82-9"><a href="visualizing-taxonomic-data.html#cb82-9" tabindex="-1"></a><span class="co">#save the files in .csv format for your analysis</span></span>
<span id="cb82-10"><a href="visualizing-taxonomic-data.html#cb82-10" tabindex="-1"></a><span class="fu">write.csv</span>(taxasums_birds, <span class="st">&quot;taxa_counts_birds_genera.csv&quot;</span>)</span>
<span id="cb82-11"><a href="visualizing-taxonomic-data.html#cb82-11" tabindex="-1"></a><span class="fu">write.csv</span>(taxasums_sediments, <span class="st">&quot;taxa_counts_sediments_genera.csv&quot;</span>)</span>
<span id="cb82-12"><a href="visualizing-taxonomic-data.html#cb82-12" tabindex="-1"></a></span>
<span id="cb82-13"><a href="visualizing-taxonomic-data.html#cb82-13" tabindex="-1"></a><span class="co">#you can specify what metadata columns you want to use like this</span></span>
<span id="cb82-14"><a href="visualizing-taxonomic-data.html#cb82-14" tabindex="-1"></a>species <span class="ot">&lt;-</span> bird_metadata<span class="sc">$</span>species</span>
<span id="cb82-15"><a href="visualizing-taxonomic-data.html#cb82-15" tabindex="-1"></a>site <span class="ot">&lt;-</span> sediment_metadata<span class="sc">$</span>site</span></code></pre></div>
<p>Now that we’re all set up, let’s create a taxonomic heatmap using the <code>taxa_heatmap</code> command for all of the seabird fecal samples. We start by using the summarized taxa file that was created above (<code>taxasums_birds</code>), and then specify the accompanying metadata file (<code>bird_metadata</code>) to be used, with the specific column we want to use for the x-axis of the heatmap (<code>species</code>). Next, we want to plot 50 taxa (<code>ntoplot=50</code>), and normalize the data with a <code>log10(percent)</code>, which emphasizes relative differences between taxa while reducing the influence of extreme values, making it easier to visualize patterns and compare abundance across taxa. Before running this command make sure you have enough space in your <code>Plots</code> quadrant on your screen has enough space to accommodate the heatmap.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="visualizing-taxonomic-data.html#cb83-1" tabindex="-1"></a><span class="fu">taxa_heatmap</span>(taxasums_birds, bird_metadata, <span class="st">&quot;species&quot;</span>, <span class="at">ntoplot=</span><span class="dv">50</span>, <span class="at">normalize =</span> <span class="st">&quot;log10(percent)&quot;</span>) <span class="sc">+</span></span>
<span id="cb83-2"><a href="visualizing-taxonomic-data.html#cb83-2" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low=</span><span class="st">&quot;#9ac094&quot;</span>, <span class="at">mid=</span><span class="st">&quot;#5ba14f&quot;</span>, <span class="at">high=</span><span class="st">&quot;#003510&quot;</span>, <span class="at">midpoint=</span><span class="dv">0</span>, <span class="at">na.value=</span><span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb83-3"><a href="visualizing-taxonomic-data.html#cb83-3" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">str_replace</span>(x, <span class="st">&quot;.*;&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">+</span></span>
<span id="cb83-4"><a href="visualizing-taxonomic-data.html#cb83-4" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span><span class="sc">-</span><span class="dv">75</span>))</span></code></pre></div>
<p>Make sure you have the <code>Plot</code> tab for the bottom right quadrant completely expanded to accommodate the size of the plot. After expanding the quadrant and running this code, you’ll notice that the y-axis takes up most of the space on the plot. Here is what the y-axis labels look like right now:</p>
<p><img src="images/heatmap_screenshot.png" /><!-- --></p>
<p>If you do not want to include the entire taxonomic lineage in the y-axis of the figure, you can simply add the <code>scale_y_discrete</code> argument to your command. This line of code modifies the y-axis labels of a discrete scale in a plot by removing everything before and including the last semicolon in each label using the str_replace function from the stringr package.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="visualizing-taxonomic-data.html#cb84-1" tabindex="-1"></a><span class="fu">taxa_heatmap</span>(taxasums_birds, bird_metadata, <span class="st">&quot;species&quot;</span>, <span class="at">ntoplot=</span><span class="dv">50</span>, <span class="at">normalize =</span> <span class="st">&quot;log10(percent)&quot;</span>) <span class="sc">+</span></span>
<span id="cb84-2"><a href="visualizing-taxonomic-data.html#cb84-2" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low=</span><span class="st">&quot;#9ac094&quot;</span>, <span class="at">mid=</span><span class="st">&quot;#5ba14f&quot;</span>, <span class="at">high=</span><span class="st">&quot;#003510&quot;</span>, <span class="at">midpoint=</span><span class="dv">0</span>, <span class="at">na.value=</span><span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb84-3"><a href="visualizing-taxonomic-data.html#cb84-3" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">str_replace</span>(x, <span class="st">&quot;.*;&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">+</span></span>
<span id="cb84-4"><a href="visualizing-taxonomic-data.html#cb84-4" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span><span class="sc">-</span><span class="dv">75</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:fig2"></span>
<img src="images/taxa_heatmap_birds.png" alt="Raw version of the taxa heatmap showing the top 50 most abundant genera present in each seabird fecal sampling group"  />
<p class="caption">
Figure 9.2: Raw version of the taxa heatmap showing the top 50 most abundant genera present in each seabird fecal sampling group
</p>
</div>
<p>Let’s do the same with the sediment samples:</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="visualizing-taxonomic-data.html#cb85-1" tabindex="-1"></a><span class="fu">taxa_heatmap</span>(taxasums_sediments, sediment_metadata, <span class="st">&quot;site&quot;</span>, <span class="at">ntoplot=</span><span class="dv">50</span>, <span class="at">normalize =</span> <span class="st">&quot;log10(percent)&quot;</span>) <span class="sc">+</span></span>
<span id="cb85-2"><a href="visualizing-taxonomic-data.html#cb85-2" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low=</span><span class="st">&quot;#c99884&quot;</span>, <span class="at">mid=</span><span class="st">&quot;#644133&quot;</span>, <span class="at">high=</span><span class="st">&quot;#351b10&quot;</span>, <span class="at">midpoint=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb85-3"><a href="visualizing-taxonomic-data.html#cb85-3" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">str_replace</span>(x, <span class="st">&quot;.*;&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">+</span></span>
<span id="cb85-4"><a href="visualizing-taxonomic-data.html#cb85-4" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>, <span class="at">angle=</span><span class="sc">-</span><span class="dv">75</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:fig3"></span>
<img src="images/taxa_heatmap_sediments.png" alt="Raw version of the taxa heatmap showing the top 50 most abundant genera present in each sediment coring site group"  />
<p class="caption">
Figure 9.3: Raw version of the taxa heatmap showing the top 50 most abundant genera present in each sediment coring site group
</p>
</div>
</div>
<div id="plot-beta-diversity" class="section level3 hasAnchor" number="9.3.7">
<h3><span class="header-section-number">9.3.7</span> Plot beta-diversity<a href="visualizing-taxonomic-data.html#plot-beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>PCA (Principal Component Analysis) plots are useful in microbiome analysis because they help visualize complex data by reducing it to a few key dimensions. In the context of comparing microbiomes between sites or species, PCA plots allow us to see patterns and similarities in microbial community composition, making it easier to identify which factors or variables are driving the differences in a data set.</p>
<p>While QIIME2 produces some nice interactive PCA plots, it’s difficult to use these for publishing. With the QIIME2R package, you can create beautiful PCA plots and edit them to your liking.</p>
<p>Start by assigning the term <code>species</code> and <code>site</code> to the appropriate column from your <code>metadata-birds.csv</code> and <code>metadata-sediments.csv</code> file:</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="visualizing-taxonomic-data.html#cb86-1" tabindex="-1"></a>species <span class="ot">&lt;-</span> bird_metadata<span class="sc">$</span>species</span>
<span id="cb86-2"><a href="visualizing-taxonomic-data.html#cb86-2" tabindex="-1"></a>site <span class="ot">&lt;-</span> sediment_metadata<span class="sc">$</span>site</span></code></pre></div>
<p>After running the <code>qiime diversity core-metrics-phylogenetic</code> command in Ch07 (Diversity Analyses), two files called <code>unweighted_unifrac_pcoa_results.qza</code> and <code>weighted_unifrac_pcoa_results.qza</code> were produced for both the sediment and seabird fecal samples (4 files total). These are the files we will use to create the weighted and unweighted PCA plots.</p>
<blockquote>
<p><strong>NOTE:</strong> Weighted UniFrac in PCA analysis takes into account the abundance of organisms, highlighting the influence of the most abundant species, while unweighted UniFrac focuses on presence and absence, emphasizing less abundant organisms in community comparisons.</p>
</blockquote>
<p>Check that these files are in the appropriate <code>/diversity</code> folders within your working directory, go back to the <strong>Exporting data</strong> section of this chapter for a quick refresher. Start by loading these files into R, where <code>wunifrac</code> refers to the weighted PCA files, and <code>unwunifrac</code> refers to the unweighted PCA files:</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="visualizing-taxonomic-data.html#cb87-1" tabindex="-1"></a>wunifrac_birds<span class="ot">&lt;-</span><span class="fu">read_qza</span>(<span class="st">&quot;diversity-birds/weighted_unifrac_pcoa_results.qza&quot;</span>)</span>
<span id="cb87-2"><a href="visualizing-taxonomic-data.html#cb87-2" tabindex="-1"></a></span>
<span id="cb87-3"><a href="visualizing-taxonomic-data.html#cb87-3" tabindex="-1"></a>unwunifrac_birds<span class="ot">&lt;-</span><span class="fu">read_qza</span>(<span class="st">&quot;diversity-birds/unweighted_unifrac_pcoa_results.qza&quot;</span>)</span>
<span id="cb87-4"><a href="visualizing-taxonomic-data.html#cb87-4" tabindex="-1"></a></span>
<span id="cb87-5"><a href="visualizing-taxonomic-data.html#cb87-5" tabindex="-1"></a>wunifrac_seds<span class="ot">&lt;-</span><span class="fu">read_qza</span>(<span class="st">&quot;diversity-sediments/weighted_unifrac_pcoa_results.qza&quot;</span>)</span>
<span id="cb87-6"><a href="visualizing-taxonomic-data.html#cb87-6" tabindex="-1"></a></span>
<span id="cb87-7"><a href="visualizing-taxonomic-data.html#cb87-7" tabindex="-1"></a>unwunifrac_seds<span class="ot">&lt;-</span><span class="fu">read_qza</span>(<span class="st">&quot;diversity-sediments/unweighted_unifrac_pcoa_results.qza&quot;</span>)</span></code></pre></div>
<p>If you want to add a little extra addition to your PCA plots, you can integrate the shannon diversity metrics into the plot. The addition of the Shannon data (shannon) in this code allows you to include the Shannon Diversity Index as a size aesthetic in the PCA plot. By doing so, you can visualize how the diversity of microbial communities (as measured by Shannon Diversity) relates to the distribution of samples in the PCA space. This can help you explore whether samples with higher or lower Shannon diversity values cluster differently in the PCA plot, providing insights into the relationship between diversity and community composition.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="visualizing-taxonomic-data.html#cb88-1" tabindex="-1"></a><span class="co"># Shannon entropy data for sediment core samples</span></span>
<span id="cb88-2"><a href="visualizing-taxonomic-data.html#cb88-2" tabindex="-1"></a>shannon_seds<span class="ot">&lt;-</span><span class="fu">read_qza</span>(<span class="st">&quot;diversity-sediments/shannon_vector.qza&quot;</span>)<span class="sc">$</span>data <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb88-3"><a href="visualizing-taxonomic-data.html#cb88-3" tabindex="-1"></a></span>
<span id="cb88-4"><a href="visualizing-taxonomic-data.html#cb88-4" tabindex="-1"></a>shannon_values_seds <span class="ot">&lt;-</span> shannon_seds<span class="sc">$</span>shannon_entropy</span>
<span id="cb88-5"><a href="visualizing-taxonomic-data.html#cb88-5" tabindex="-1"></a></span>
<span id="cb88-6"><a href="visualizing-taxonomic-data.html#cb88-6" tabindex="-1"></a><span class="co"># Shannon entropy data for seabird fecal samples</span></span>
<span id="cb88-7"><a href="visualizing-taxonomic-data.html#cb88-7" tabindex="-1"></a>shannon_birds<span class="ot">&lt;-</span><span class="fu">read_qza</span>(<span class="st">&quot;diversity-birds/shannon_vector.qza&quot;</span>)<span class="sc">$</span>data <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb88-8"><a href="visualizing-taxonomic-data.html#cb88-8" tabindex="-1"></a></span>
<span id="cb88-9"><a href="visualizing-taxonomic-data.html#cb88-9" tabindex="-1"></a>shannon_values_birds <span class="ot">&lt;-</span> shannon_birds<span class="sc">$</span>shannon_entropy</span></code></pre></div>
<p>The next commands are used to plot the weighted (<code>PCA_w</code>) and unweighted (<code>PCA_unw</code>)PCA’s of the sediment samples. This command can be broken down into several different parts that are explained below:</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="visualizing-taxonomic-data.html#cb89-1" tabindex="-1"></a><span class="co">#plotting weighted PCA for sediments</span></span>
<span id="cb89-2"><a href="visualizing-taxonomic-data.html#cb89-2" tabindex="-1"></a></span>
<span id="cb89-3"><a href="visualizing-taxonomic-data.html#cb89-3" tabindex="-1"></a>PCA_w_seds <span class="ot">&lt;-</span> wunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>Vectors <span class="sc">%&gt;%</span></span>
<span id="cb89-4"><a href="visualizing-taxonomic-data.html#cb89-4" tabindex="-1"></a>  <span class="fu">select</span>(SampleID, PC1, PC2) <span class="sc">%&gt;%</span></span>
<span id="cb89-5"><a href="visualizing-taxonomic-data.html#cb89-5" tabindex="-1"></a>  <span class="fu">left_join</span>(sediment_metadata) <span class="sc">%&gt;%</span></span>
<span id="cb89-6"><a href="visualizing-taxonomic-data.html#cb89-6" tabindex="-1"></a>  <span class="fu">left_join</span>(shannon_seds) <span class="sc">%&gt;%</span></span>
<span id="cb89-7"><a href="visualizing-taxonomic-data.html#cb89-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> site, <span class="at">size =</span> shannon_values_seds)) <span class="sc">+</span></span>
<span id="cb89-8"><a href="visualizing-taxonomic-data.html#cb89-8" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb89-9"><a href="visualizing-taxonomic-data.html#cb89-9" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="at">type =</span> <span class="st">&quot;norm&quot;</span>, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="dv">1</span>, <span class="at">geom =</span> <span class="st">&quot;polygon&quot;</span>, <span class="fu">aes</span>(<span class="at">fill =</span> site), <span class="at">alpha =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb89-10"><a href="visualizing-taxonomic-data.html#cb89-10" tabindex="-1"></a>  <span class="fu">theme_q2r</span>() <span class="sc">+</span></span>
<span id="cb89-11"><a href="visualizing-taxonomic-data.html#cb89-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="sc">+</span></span>
<span id="cb89-12"><a href="visualizing-taxonomic-data.html#cb89-12" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> wunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC1&quot;</span>)) <span class="sc">+</span></span>
<span id="cb89-13"><a href="visualizing-taxonomic-data.html#cb89-13" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> wunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC2&quot;</span>)) <span class="sc">+</span></span>
<span id="cb89-14"><a href="visualizing-taxonomic-data.html#cb89-14" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">&quot;Shannon Diversity&quot;</span>)</span>
<span id="cb89-15"><a href="visualizing-taxonomic-data.html#cb89-15" tabindex="-1"></a></span>
<span id="cb89-16"><a href="visualizing-taxonomic-data.html#cb89-16" tabindex="-1"></a></span>
<span id="cb89-17"><a href="visualizing-taxonomic-data.html#cb89-17" tabindex="-1"></a>PCA_w_seds</span>
<span id="cb89-18"><a href="visualizing-taxonomic-data.html#cb89-18" tabindex="-1"></a></span>
<span id="cb89-19"><a href="visualizing-taxonomic-data.html#cb89-19" tabindex="-1"></a></span>
<span id="cb89-20"><a href="visualizing-taxonomic-data.html#cb89-20" tabindex="-1"></a><span class="co">#plotting unweighted PCA for sediments</span></span>
<span id="cb89-21"><a href="visualizing-taxonomic-data.html#cb89-21" tabindex="-1"></a></span>
<span id="cb89-22"><a href="visualizing-taxonomic-data.html#cb89-22" tabindex="-1"></a>PCA_unw_seds <span class="ot">&lt;-</span> unwunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>Vectors <span class="sc">%&gt;%</span></span>
<span id="cb89-23"><a href="visualizing-taxonomic-data.html#cb89-23" tabindex="-1"></a>  <span class="fu">select</span>(SampleID, PC1, PC2) <span class="sc">%&gt;%</span></span>
<span id="cb89-24"><a href="visualizing-taxonomic-data.html#cb89-24" tabindex="-1"></a>  <span class="fu">left_join</span>(sediment_metadata) <span class="sc">%&gt;%</span></span>
<span id="cb89-25"><a href="visualizing-taxonomic-data.html#cb89-25" tabindex="-1"></a>  <span class="fu">left_join</span>(shannon_seds) <span class="sc">%&gt;%</span></span>
<span id="cb89-26"><a href="visualizing-taxonomic-data.html#cb89-26" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> site, <span class="at">size =</span> shannon_values_seds)) <span class="sc">+</span></span>
<span id="cb89-27"><a href="visualizing-taxonomic-data.html#cb89-27" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb89-28"><a href="visualizing-taxonomic-data.html#cb89-28" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="at">type =</span> <span class="st">&quot;norm&quot;</span>, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="dv">12</span>, <span class="at">geom =</span> <span class="st">&quot;polygon&quot;</span>, <span class="fu">aes</span>(<span class="at">fill =</span> site), <span class="at">alpha =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb89-29"><a href="visualizing-taxonomic-data.html#cb89-29" tabindex="-1"></a>  <span class="fu">theme_q2r</span>() <span class="sc">+</span></span>
<span id="cb89-30"><a href="visualizing-taxonomic-data.html#cb89-30" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="sc">+</span></span>
<span id="cb89-31"><a href="visualizing-taxonomic-data.html#cb89-31" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> unwunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC1&quot;</span>)) <span class="sc">+</span></span>
<span id="cb89-32"><a href="visualizing-taxonomic-data.html#cb89-32" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> unwunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC2&quot;</span>)) <span class="sc">+</span></span>
<span id="cb89-33"><a href="visualizing-taxonomic-data.html#cb89-33" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">&quot;Shannon Diversity&quot;</span>)  <span class="co"># Set the legend title for size</span></span>
<span id="cb89-34"><a href="visualizing-taxonomic-data.html#cb89-34" tabindex="-1"></a></span>
<span id="cb89-35"><a href="visualizing-taxonomic-data.html#cb89-35" tabindex="-1"></a>PCA_unw_seds</span></code></pre></div>
<p>Here is an explanation of some of the arguments in the PCA plotting command:</p>
<ul>
<li><p><code>PCA_w_seds &lt;- wunifrac_seds$data$Vectors %&gt;%</code> : This part of the command selects the <code>Vectors</code> data from the <code>wunifrac_seds</code> object and begins a chain of operations.</p></li>
<li><p><code>select(SampleID, PC1, PC2) %&gt;%</code> : selects three columns from the data: “SampleID,” “PC1,” and “PC2.”</p></li>
</ul>
<p>*<code>left_join(sediment_metadata) %&gt;%</code>: performs a left join with the data from <code>sediment_metadata</code>, combining the selected columns from the previous step with the <code>SampleID</code> column.</p>
<ul>
<li><p><code>left_join(shannon_seds) %&gt;%</code> : similar to the previous step, performs a left join with the “shannon_seds” data, combining the result of the previous join with this data.</p></li>
<li><p><code>ggplot(aes(x = PC1, y = PC2, color = site, size = shannon_values_seds))</code> : This part starts creating a ggplot, where the <code>color</code> aesthetic is determined by <code>site</code>, and the <code>size</code> aesthetic is determined by <code>shannon_values_seds</code>.</p></li>
<li><p><code>stat_ellipse(type = "norm", level = 0.95, linetype = 1, geom = "polygon", aes(fill = site), alpha = 0.05)</code>: This adds ellipses to the plot that represent normal distribution-based confidence intervals (at 95% level) around the points. I chose linetype 1 for weighted PCAs and linetype 12 for unweighted PCAs.</p></li>
<li><p><code>xlab(paste(round(100 * wunifrac_seds$data$ProportionExplained[1], 2), "% - PC1"))</code> : Sets the x-axis label (and y-axis label in the next line) using the info from the “wunifrac_seds” object that denotes the proportion explained by the first principal component (PC1, or PC2 for the y-axis), converting it to a percentage.</p></li>
</ul>
<p>Now plot the same for the seabird fecal samples:</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="visualizing-taxonomic-data.html#cb90-1" tabindex="-1"></a><span class="co">#plotting weighted PCA for seabird fecal samples</span></span>
<span id="cb90-2"><a href="visualizing-taxonomic-data.html#cb90-2" tabindex="-1"></a></span>
<span id="cb90-3"><a href="visualizing-taxonomic-data.html#cb90-3" tabindex="-1"></a>PCA_w_birds <span class="ot">&lt;-</span> wunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>Vectors <span class="sc">%&gt;%</span></span>
<span id="cb90-4"><a href="visualizing-taxonomic-data.html#cb90-4" tabindex="-1"></a>  <span class="fu">select</span>(SampleID, PC1, PC2) <span class="sc">%&gt;%</span></span>
<span id="cb90-5"><a href="visualizing-taxonomic-data.html#cb90-5" tabindex="-1"></a>  <span class="fu">left_join</span>(bird_metadata) <span class="sc">%&gt;%</span></span>
<span id="cb90-6"><a href="visualizing-taxonomic-data.html#cb90-6" tabindex="-1"></a>  <span class="fu">left_join</span>(shannon_birds) <span class="sc">%&gt;%</span></span>
<span id="cb90-7"><a href="visualizing-taxonomic-data.html#cb90-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> species, <span class="at">size =</span> shannon_values_birds)) <span class="sc">+</span></span>
<span id="cb90-8"><a href="visualizing-taxonomic-data.html#cb90-8" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb90-9"><a href="visualizing-taxonomic-data.html#cb90-9" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="at">type =</span> <span class="st">&quot;norm&quot;</span>, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="dv">1</span>, <span class="at">geom =</span> <span class="st">&quot;polygon&quot;</span>, <span class="fu">aes</span>(<span class="at">fill =</span> species), <span class="at">alpha =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb90-10"><a href="visualizing-taxonomic-data.html#cb90-10" tabindex="-1"></a>  <span class="fu">theme_q2r</span>() <span class="sc">+</span></span>
<span id="cb90-11"><a href="visualizing-taxonomic-data.html#cb90-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="sc">+</span></span>
<span id="cb90-12"><a href="visualizing-taxonomic-data.html#cb90-12" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> wunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC1&quot;</span>)) <span class="sc">+</span></span>
<span id="cb90-13"><a href="visualizing-taxonomic-data.html#cb90-13" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> wunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC2&quot;</span>)) <span class="sc">+</span></span>
<span id="cb90-14"><a href="visualizing-taxonomic-data.html#cb90-14" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">&quot;Shannon Diversity&quot;</span>)  <span class="co"># Set the legend title for size</span></span>
<span id="cb90-15"><a href="visualizing-taxonomic-data.html#cb90-15" tabindex="-1"></a></span>
<span id="cb90-16"><a href="visualizing-taxonomic-data.html#cb90-16" tabindex="-1"></a></span>
<span id="cb90-17"><a href="visualizing-taxonomic-data.html#cb90-17" tabindex="-1"></a>PCA_w_birds</span>
<span id="cb90-18"><a href="visualizing-taxonomic-data.html#cb90-18" tabindex="-1"></a></span>
<span id="cb90-19"><a href="visualizing-taxonomic-data.html#cb90-19" tabindex="-1"></a></span>
<span id="cb90-20"><a href="visualizing-taxonomic-data.html#cb90-20" tabindex="-1"></a><span class="co">#plotting weighted PCA for seabird fecal samples</span></span>
<span id="cb90-21"><a href="visualizing-taxonomic-data.html#cb90-21" tabindex="-1"></a></span>
<span id="cb90-22"><a href="visualizing-taxonomic-data.html#cb90-22" tabindex="-1"></a>PCA_unw_birds <span class="ot">&lt;-</span> unwunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>Vectors <span class="sc">%&gt;%</span></span>
<span id="cb90-23"><a href="visualizing-taxonomic-data.html#cb90-23" tabindex="-1"></a>  <span class="fu">select</span>(SampleID, PC1, PC2) <span class="sc">%&gt;%</span></span>
<span id="cb90-24"><a href="visualizing-taxonomic-data.html#cb90-24" tabindex="-1"></a>  <span class="fu">left_join</span>(bird_metadata) <span class="sc">%&gt;%</span></span>
<span id="cb90-25"><a href="visualizing-taxonomic-data.html#cb90-25" tabindex="-1"></a>  <span class="fu">left_join</span>(shannon_birds) <span class="sc">%&gt;%</span></span>
<span id="cb90-26"><a href="visualizing-taxonomic-data.html#cb90-26" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> species, <span class="at">size =</span> shannon_values_birds)) <span class="sc">+</span></span>
<span id="cb90-27"><a href="visualizing-taxonomic-data.html#cb90-27" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb90-28"><a href="visualizing-taxonomic-data.html#cb90-28" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="at">type =</span> <span class="st">&quot;norm&quot;</span>, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="dv">12</span>, <span class="at">geom =</span> <span class="st">&quot;polygon&quot;</span>, <span class="fu">aes</span>(<span class="at">fill =</span> species), <span class="at">alpha =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb90-29"><a href="visualizing-taxonomic-data.html#cb90-29" tabindex="-1"></a>  <span class="fu">theme_q2r</span>() <span class="sc">+</span></span>
<span id="cb90-30"><a href="visualizing-taxonomic-data.html#cb90-30" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="sc">+</span></span>
<span id="cb90-31"><a href="visualizing-taxonomic-data.html#cb90-31" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> unwunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC1&quot;</span>)) <span class="sc">+</span></span>
<span id="cb90-32"><a href="visualizing-taxonomic-data.html#cb90-32" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> unwunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC2&quot;</span>)) <span class="sc">+</span></span>
<span id="cb90-33"><a href="visualizing-taxonomic-data.html#cb90-33" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">&quot;Shannon Diversity&quot;</span>)  <span class="co"># Set the legend title for size</span></span>
<span id="cb90-34"><a href="visualizing-taxonomic-data.html#cb90-34" tabindex="-1"></a></span>
<span id="cb90-35"><a href="visualizing-taxonomic-data.html#cb90-35" tabindex="-1"></a>PCA_unw_birds</span></code></pre></div>
<p>Now that we have all 4 plots done for both unweighted and weighted metrics, lets put them all on the same figure using the <code>patchwork</code> package provided by R.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="visualizing-taxonomic-data.html#cb91-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;patchwork&quot;</span>)</span>
<span id="cb91-2"><a href="visualizing-taxonomic-data.html#cb91-2" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb91-3"><a href="visualizing-taxonomic-data.html#cb91-3" tabindex="-1"></a><span class="fu">wrap_plots</span>(PCA_w_seds, PCA_unw_seds, PCA_w_birds, PCA_unw_birds)</span></code></pre></div>
<p>Make sure you expand your bottom left quadrant in RStudio as much as possible to accommodate the plot size, as you can see in this screenshot. Your wrapped PCA figure will look something like this:
<img src="images/PCA_1.png" /><!-- --></p>
<p>You can see that we don’t really need two legends for both unweighted and weighted versions of the same plot. Let’s quickly fix the code to remove a legend from the two weighted plots to the left, so that all we have is legends to the right hand side of this figure by adding <code>legend.position="none"</code> to the <code>theme()</code> argument we already have going.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="visualizing-taxonomic-data.html#cb92-1" tabindex="-1"></a><span class="co"># removing the legend for the sediment sample&#39;s weighted PCA</span></span>
<span id="cb92-2"><a href="visualizing-taxonomic-data.html#cb92-2" tabindex="-1"></a></span>
<span id="cb92-3"><a href="visualizing-taxonomic-data.html#cb92-3" tabindex="-1"></a>PCA_w_seds <span class="ot">&lt;-</span> wunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>Vectors <span class="sc">%&gt;%</span></span>
<span id="cb92-4"><a href="visualizing-taxonomic-data.html#cb92-4" tabindex="-1"></a>  <span class="fu">select</span>(SampleID, PC1, PC2) <span class="sc">%&gt;%</span></span>
<span id="cb92-5"><a href="visualizing-taxonomic-data.html#cb92-5" tabindex="-1"></a>  <span class="fu">left_join</span>(sediment_metadata) <span class="sc">%&gt;%</span></span>
<span id="cb92-6"><a href="visualizing-taxonomic-data.html#cb92-6" tabindex="-1"></a>  <span class="fu">left_join</span>(shannon_seds) <span class="sc">%&gt;%</span></span>
<span id="cb92-7"><a href="visualizing-taxonomic-data.html#cb92-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> site, <span class="at">size =</span> shannon_values_seds)) <span class="sc">+</span></span>
<span id="cb92-8"><a href="visualizing-taxonomic-data.html#cb92-8" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb92-9"><a href="visualizing-taxonomic-data.html#cb92-9" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="at">type =</span> <span class="st">&quot;norm&quot;</span>, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="dv">1</span>, <span class="at">geom =</span> <span class="st">&quot;polygon&quot;</span>, <span class="fu">aes</span>(<span class="at">fill =</span> site), <span class="at">alpha =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb92-10"><a href="visualizing-taxonomic-data.html#cb92-10" tabindex="-1"></a>  <span class="fu">theme_q2r</span>() <span class="sc">+</span></span>
<span id="cb92-11"><a href="visualizing-taxonomic-data.html#cb92-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb92-12"><a href="visualizing-taxonomic-data.html#cb92-12" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> wunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC1&quot;</span>)) <span class="sc">+</span></span>
<span id="cb92-13"><a href="visualizing-taxonomic-data.html#cb92-13" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> wunifrac_seds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC2&quot;</span>)) <span class="sc">+</span></span>
<span id="cb92-14"><a href="visualizing-taxonomic-data.html#cb92-14" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">&quot;Shannon Diversity&quot;</span>)</span>
<span id="cb92-15"><a href="visualizing-taxonomic-data.html#cb92-15" tabindex="-1"></a></span>
<span id="cb92-16"><a href="visualizing-taxonomic-data.html#cb92-16" tabindex="-1"></a><span class="co"># removing the legend for the seabird fecal sample&#39;s weighted PCA</span></span>
<span id="cb92-17"><a href="visualizing-taxonomic-data.html#cb92-17" tabindex="-1"></a></span>
<span id="cb92-18"><a href="visualizing-taxonomic-data.html#cb92-18" tabindex="-1"></a>PCA_w_birds <span class="ot">&lt;-</span> wunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>Vectors <span class="sc">%&gt;%</span></span>
<span id="cb92-19"><a href="visualizing-taxonomic-data.html#cb92-19" tabindex="-1"></a>  <span class="fu">select</span>(SampleID, PC1, PC2) <span class="sc">%&gt;%</span></span>
<span id="cb92-20"><a href="visualizing-taxonomic-data.html#cb92-20" tabindex="-1"></a>  <span class="fu">left_join</span>(bird_metadata) <span class="sc">%&gt;%</span></span>
<span id="cb92-21"><a href="visualizing-taxonomic-data.html#cb92-21" tabindex="-1"></a>  <span class="fu">left_join</span>(shannon_birds) <span class="sc">%&gt;%</span></span>
<span id="cb92-22"><a href="visualizing-taxonomic-data.html#cb92-22" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> species, <span class="at">size =</span> shannon_values_birds)) <span class="sc">+</span></span>
<span id="cb92-23"><a href="visualizing-taxonomic-data.html#cb92-23" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb92-24"><a href="visualizing-taxonomic-data.html#cb92-24" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="at">type =</span> <span class="st">&quot;norm&quot;</span>, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">linetype =</span> <span class="dv">1</span>, <span class="at">geom =</span> <span class="st">&quot;polygon&quot;</span>, <span class="fu">aes</span>(<span class="at">fill =</span> species), <span class="at">alpha =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb92-25"><a href="visualizing-taxonomic-data.html#cb92-25" tabindex="-1"></a>  <span class="fu">theme_q2r</span>() <span class="sc">+</span></span>
<span id="cb92-26"><a href="visualizing-taxonomic-data.html#cb92-26" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_line</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb92-27"><a href="visualizing-taxonomic-data.html#cb92-27" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> wunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC1&quot;</span>)) <span class="sc">+</span></span>
<span id="cb92-28"><a href="visualizing-taxonomic-data.html#cb92-28" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> wunifrac_birds<span class="sc">$</span>data<span class="sc">$</span>ProportionExplained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;% - PC2&quot;</span>)) <span class="sc">+</span></span>
<span id="cb92-29"><a href="visualizing-taxonomic-data.html#cb92-29" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">&quot;Shannon Diversity&quot;</span>)  <span class="co"># Set the legend title for size</span></span></code></pre></div>
<p>Now wrap the plots one more time to see the final result!</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="visualizing-taxonomic-data.html#cb93-1" tabindex="-1"></a><span class="fu">wrap_plots</span>(PCA_w_seds, PCA_unw_seds, PCA_w_birds, PCA_unw_birds)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:fig4"></span>
<img src="images/PCA_2.png" alt="Raw version of the weighted PCA (two left plots) and unweighted PCA (two right plots) for both sediment samples (top) and seabird fecal samples (bottom)"  />
<p class="caption">
Figure 9.4: Raw version of the weighted PCA (two left plots) and unweighted PCA (two right plots) for both sediment samples (top) and seabird fecal samples (bottom)
</p>
</div>
</div>
</div>
<div id="phyloseq" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> <strong>Phyloseq</strong><a href="visualizing-taxonomic-data.html#phyloseq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://joey711.github.io/phyloseq/">Phyloseq</a> is a handy tool in microbiome analysis that helps researchers work with and visualize microbial community data <strong>[19]</strong>. It can be used alongside QIIME2 artifact files by converting QIIME2 outputs into a Phyloseq object. Like QIIME2R, it allows for advanced analysis, visualization, and exploration of microbiome data within R’s programming environment.</p>
<p>install and load the needed packages</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="visualizing-taxonomic-data.html#cb94-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb94-2"><a href="visualizing-taxonomic-data.html#cb94-2" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb94-3"><a href="visualizing-taxonomic-data.html#cb94-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb94-4"><a href="visualizing-taxonomic-data.html#cb94-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb94-5"><a href="visualizing-taxonomic-data.html#cb94-5" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span></code></pre></div>
<div id="build-a-phyloseq-object" class="section level3 hasAnchor" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> Build a phyloseq object<a href="visualizing-taxonomic-data.html#build-a-phyloseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, create a phyloseq object for your data. These objects can contain several different .QZA files. In this case, they include the feature table, taxonomy file and metadata file, but they can also include the trees, insert-placements and other files which were created throughout the preprocessing and diversity analysis stages.</p>
<p>In these next commands, I build 2 phyloseq objects for (1) the seabird fecal samples, (2) the sediment samples. Make sure you are in the correct working directory, in this case <code>/QIIME_files</code>.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="visualizing-taxonomic-data.html#cb95-1" tabindex="-1"></a><span class="co">#building an object for seabird fecal samples</span></span>
<span id="cb95-2"><a href="visualizing-taxonomic-data.html#cb95-2" tabindex="-1"></a>birds_physeq<span class="ot">&lt;-</span><span class="fu">qza_to_phyloseq</span>(</span>
<span id="cb95-3"><a href="visualizing-taxonomic-data.html#cb95-3" tabindex="-1"></a>  <span class="at">features=</span><span class="st">&quot;final_table-birds.qza&quot;</span>,</span>
<span id="cb95-4"><a href="visualizing-taxonomic-data.html#cb95-4" tabindex="-1"></a>  <span class="at">tree=</span><span class="st">&quot;asvs-tree_birds.qza&quot;</span>,<span class="st">&quot;classification_birds.qza&quot;</span>,</span>
<span id="cb95-5"><a href="visualizing-taxonomic-data.html#cb95-5" tabindex="-1"></a>  <span class="at">metadata =</span> <span class="st">&quot;metadata-birds.csv&quot;</span>)</span>
<span id="cb95-6"><a href="visualizing-taxonomic-data.html#cb95-6" tabindex="-1"></a></span>
<span id="cb95-7"><a href="visualizing-taxonomic-data.html#cb95-7" tabindex="-1"></a>birds_physeq</span>
<span id="cb95-8"><a href="visualizing-taxonomic-data.html#cb95-8" tabindex="-1"></a></span>
<span id="cb95-9"><a href="visualizing-taxonomic-data.html#cb95-9" tabindex="-1"></a><span class="co">#building an object for sediment core samples</span></span>
<span id="cb95-10"><a href="visualizing-taxonomic-data.html#cb95-10" tabindex="-1"></a>sediments_physeq<span class="ot">&lt;-</span><span class="fu">qza_to_phyloseq</span>(</span>
<span id="cb95-11"><a href="visualizing-taxonomic-data.html#cb95-11" tabindex="-1"></a>  <span class="at">features=</span><span class="st">&quot;final_table-sediments.qza&quot;</span>,</span>
<span id="cb95-12"><a href="visualizing-taxonomic-data.html#cb95-12" tabindex="-1"></a>  <span class="at">tree=</span><span class="st">&quot;asvs-tree_sediments.qza&quot;</span>,<span class="st">&quot;classification_sediments.qza&quot;</span>,</span>
<span id="cb95-13"><a href="visualizing-taxonomic-data.html#cb95-13" tabindex="-1"></a>  <span class="at">metadata =</span> <span class="st">&quot;metadata-sediments.csv&quot;</span>)</span>
<span id="cb95-14"><a href="visualizing-taxonomic-data.html#cb95-14" tabindex="-1"></a></span>
<span id="cb95-15"><a href="visualizing-taxonomic-data.html#cb95-15" tabindex="-1"></a>sediments_physeq</span></code></pre></div>
</div>
<div id="taxonomic-characteristics" class="section level3 hasAnchor" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> Taxonomic characteristics<a href="visualizing-taxonomic-data.html#taxonomic-characteristics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If you want to find out how many taxa have been classified at different taxonomic levels within these objects, use the following commands:</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="visualizing-taxonomic-data.html#cb96-1" tabindex="-1"></a><span class="co">#for seabird fecal samples</span></span>
<span id="cb96-2"><a href="visualizing-taxonomic-data.html#cb96-2" tabindex="-1"></a><span class="fu">tax_table</span>(birds_physeq) <span class="sc">%&gt;%</span></span>
<span id="cb96-3"><a href="visualizing-taxonomic-data.html#cb96-3" tabindex="-1"></a>  <span class="fu">as</span>(<span class="st">&quot;matrix&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-4"><a href="visualizing-taxonomic-data.html#cb96-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">&quot;OTU&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-5"><a href="visualizing-taxonomic-data.html#cb96-5" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="st">&quot;Rank&quot;</span>, <span class="st">&quot;Name&quot;</span>, <span class="fu">rank_names</span>(birds_physeq)) <span class="sc">%&gt;%</span></span>
<span id="cb96-6"><a href="visualizing-taxonomic-data.html#cb96-6" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> <span class="co"># remove rows with NA value</span></span>
<span id="cb96-7"><a href="visualizing-taxonomic-data.html#cb96-7" tabindex="-1"></a>  <span class="fu">group_by</span>(Rank) <span class="sc">%&gt;%</span></span>
<span id="cb96-8"><a href="visualizing-taxonomic-data.html#cb96-8" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">ntaxa =</span> <span class="fu">length</span>(<span class="fu">unique</span>(Name))) <span class="sc">%&gt;%</span> <span class="co"># computes number of unique taxa</span></span>
<span id="cb96-9"><a href="visualizing-taxonomic-data.html#cb96-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Rank =</span> <span class="fu">factor</span>(Rank, <span class="fu">rank_names</span>(birds_physeq))) <span class="sc">%&gt;%</span></span>
<span id="cb96-10"><a href="visualizing-taxonomic-data.html#cb96-10" tabindex="-1"></a>  <span class="fu">arrange</span>(Rank) </span>
<span id="cb96-11"><a href="visualizing-taxonomic-data.html#cb96-11" tabindex="-1"></a></span>
<span id="cb96-12"><a href="visualizing-taxonomic-data.html#cb96-12" tabindex="-1"></a><span class="co">#for sediment samples</span></span>
<span id="cb96-13"><a href="visualizing-taxonomic-data.html#cb96-13" tabindex="-1"></a><span class="fu">tax_table</span>(sediments_physeq) <span class="sc">%&gt;%</span></span>
<span id="cb96-14"><a href="visualizing-taxonomic-data.html#cb96-14" tabindex="-1"></a>  <span class="fu">as</span>(<span class="st">&quot;matrix&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-15"><a href="visualizing-taxonomic-data.html#cb96-15" tabindex="-1"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">&quot;OTU&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-16"><a href="visualizing-taxonomic-data.html#cb96-16" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="st">&quot;Rank&quot;</span>, <span class="st">&quot;Name&quot;</span>, <span class="fu">rank_names</span>(sediments_physeq)) <span class="sc">%&gt;%</span></span>
<span id="cb96-17"><a href="visualizing-taxonomic-data.html#cb96-17" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb96-18"><a href="visualizing-taxonomic-data.html#cb96-18" tabindex="-1"></a>  <span class="fu">group_by</span>(Rank) <span class="sc">%&gt;%</span></span>
<span id="cb96-19"><a href="visualizing-taxonomic-data.html#cb96-19" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">ntaxa =</span> <span class="fu">length</span>(<span class="fu">unique</span>(Name))) <span class="sc">%&gt;%</span></span>
<span id="cb96-20"><a href="visualizing-taxonomic-data.html#cb96-20" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Rank =</span> <span class="fu">factor</span>(Rank, <span class="fu">rank_names</span>(sediments_physeq))) <span class="sc">%&gt;%</span></span>
<span id="cb96-21"><a href="visualizing-taxonomic-data.html#cb96-21" tabindex="-1"></a>  <span class="fu">arrange</span>(Rank)</span></code></pre></div>
<p>The above command takes the taxonomic table in the <code>birds_physeq</code> object and performs a series of data manipulation operations, using the dplyr and tidyr packages in R. It first converts the taxonomic table to a matrix, then converts it to a tibble with row names set as <code>OTU</code> (used as a generic term here even though I am using ASVs). It gathers the data to create two columns, <code>Rank</code> and <code>Name</code>, from the taxonomic ranks present in <code>birds_physeq</code>. The<code>na.omit()</code> line removes rows with missing values, and then <code>group_by()</code> line groups the data by <code>Rank</code>, then calculates the number of unique taxa for each rank using the <code>summarize()</code> command. Finally, the <code>mutate()</code> command converts <code>Rank</code> into a factor variable with predefined rank names, and finally arranges the data by <code>Rank</code> in ascending order using the <code>arrange()</code> command. The output for the command will look something like this (these numbers are from running the command on the sediment sample phyloseq object):</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="visualizing-taxonomic-data.html#cb97-1" tabindex="-1"></a><span class="co"># A tibble: 7 × 2</span></span>
<span id="cb97-2"><a href="visualizing-taxonomic-data.html#cb97-2" tabindex="-1"></a>  Rank    ntaxa</span>
<span id="cb97-3"><a href="visualizing-taxonomic-data.html#cb97-3" tabindex="-1"></a>  <span class="sc">&lt;</span>fct<span class="sc">&gt;</span>   <span class="er">&lt;</span>int<span class="sc">&gt;</span></span>
<span id="cb97-4"><a href="visualizing-taxonomic-data.html#cb97-4" tabindex="-1"></a><span class="dv">1</span> Kingdom     <span class="dv">2</span></span>
<span id="cb97-5"><a href="visualizing-taxonomic-data.html#cb97-5" tabindex="-1"></a><span class="dv">2</span> Phylum     <span class="dv">57</span></span>
<span id="cb97-6"><a href="visualizing-taxonomic-data.html#cb97-6" tabindex="-1"></a><span class="dv">3</span> Class     <span class="dv">125</span></span>
<span id="cb97-7"><a href="visualizing-taxonomic-data.html#cb97-7" tabindex="-1"></a><span class="dv">4</span> Order     <span class="dv">244</span></span>
<span id="cb97-8"><a href="visualizing-taxonomic-data.html#cb97-8" tabindex="-1"></a><span class="dv">5</span> Family    <span class="dv">309</span></span>
<span id="cb97-9"><a href="visualizing-taxonomic-data.html#cb97-9" tabindex="-1"></a><span class="dv">6</span> Genus     <span class="dv">393</span></span>
<span id="cb97-10"><a href="visualizing-taxonomic-data.html#cb97-10" tabindex="-1"></a><span class="dv">7</span> Species   <span class="dv">160</span></span></code></pre></div>
</div>
<div id="plot-alpha-diversity" class="section level3 hasAnchor" number="9.4.3">
<h3><span class="header-section-number">9.4.3</span> Plot alpha-diversity<a href="visualizing-taxonomic-data.html#plot-alpha-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>What’s nice about Phyloseq is that you can calculate and plot diversity metrics all in one command, so you actually don’t even have to run any qiime diversity commands if you don’t want to! I like the plots that this Phyloseq package produces more anyways, and they are easier to save as an .SVG image to edit later on for publishing.</p>
<p>Here I plot three diversity metrics together for the sediment samples according to site, using the following <code>plot_richness</code> command:</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="visualizing-taxonomic-data.html#cb98-1" tabindex="-1"></a>sediments_richness <span class="ot">&lt;-</span> <span class="fu">plot_richness</span>(sediments_physeq, <span class="at">x =</span> <span class="st">&quot;site&quot;</span>, <span class="at">color=</span><span class="st">&quot;depth&quot;</span>, <span class="at">nrow=</span><span class="dv">1</span>, </span>
<span id="cb98-2"><a href="visualizing-taxonomic-data.html#cb98-2" tabindex="-1"></a>                                    <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Observed&quot;</span>)) <span class="sc">+</span></span>
<span id="cb98-3"><a href="visualizing-taxonomic-data.html#cb98-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb98-4"><a href="visualizing-taxonomic-data.html#cb98-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha=</span><span class="fl">0.3</span>, <span class="at">size=</span><span class="fl">0.3</span>, <span class="at">colour=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb98-5"><a href="visualizing-taxonomic-data.html#cb98-5" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb98-6"><a href="visualizing-taxonomic-data.html#cb98-6" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), <span class="at">panel.grid.major.x=</span><span class="fu">element_line</span>(<span class="cn">FALSE</span>), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>), <span class="at">axis.text.x=</span>(<span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">vjust=</span><span class="fl">0.5</span>)))</span></code></pre></div>
<p>The above command includes metrics for Simpson, Shannon, and observed features as subplots. I assign the plot to the term <code>sediments_richness</code> so that I can merge the diversity plots for both sediment and seabird samples together afterwards. Here is a quick review of the arguments I used here:</p>
<ul>
<li><p><code>x="site"</code> will split the plot up into the 3 coring sites listed in the metadata file of the phyloseq object</p></li>
<li><p><code>color=depth</code> will color the points on the box plot so that diversity can be visualized across different depths of the core.</p></li>
<li><p>The <code>nrow</code> argument let’s you adjust the orientation of the plots, with 1 row the plots will end up in a horizontal line, with 3 rows the plots will end up in a vertical line.</p></li>
<li><p>The <code>geom_points()</code> and <code>geom_boxplot()</code> arguments let you manipulate the size and transparency (alpha) of the box plot’s features.</p></li>
</ul>
<blockquote>
<p>When adding a theme to boxplot, there are several options that you can choose from (to see a list of the themes visit <a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">this link</a>). I chose the black and white theme (<code>theme_bw()</code>) for my plot, and then on the last line I use the <code>theme()</code> argument to specify the overall text size of the boxplot and legend, as well as the axis text size.</p>
</blockquote>
<ul>
<li><p>The <code>panel.grid.major.x=element_line(FALSE)</code> argument to removes the vertical grid lines as they aren’t necessary here.</p></li>
<li><p>The <code>panel.grid.major.y=element_line()</code> argument can also be used to specify the <code>color()</code>, <code>linetype()</code> and <code>size()</code> of your grid lines.</p></li>
<li><p>The <code>axis.text.x=(element_text(angle=90, vjust=0.5)</code> argument changes the orientation of the x-axis text so that it all fits better in the plot using the <code>angle</code> option, and the <code>vjust</code> option to adjust the vertical position of the labels so they are not overlapping the axis.</p></li>
</ul>
<p>Next, do the same for the seabird fecal samples. This time plotting by species, without any colors for the points.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="visualizing-taxonomic-data.html#cb99-1" tabindex="-1"></a>birds_richness <span class="ot">&lt;-</span> <span class="fu">plot_richness</span>(birds_physeq, <span class="at">x =</span> <span class="st">&quot;species&quot;</span>, <span class="at">nrow=</span><span class="dv">3</span>, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Observed&quot;</span>)) <span class="sc">+</span></span>
<span id="cb99-2"><a href="visualizing-taxonomic-data.html#cb99-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb99-3"><a href="visualizing-taxonomic-data.html#cb99-3" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha=</span><span class="fl">0.3</span>, <span class="at">size=</span><span class="fl">0.3</span>, <span class="at">colour=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb99-4"><a href="visualizing-taxonomic-data.html#cb99-4" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb99-5"><a href="visualizing-taxonomic-data.html#cb99-5" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), <span class="at">panel.grid.major.x=</span><span class="fu">element_line</span>(<span class="cn">FALSE</span>), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>), <span class="at">axis.text.x=</span>(<span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">vjust=</span><span class="fl">0.5</span>)))</span></code></pre></div>
<p>Now we can make a larger plot for publishing that puts all of these plots together in one figure. You can always do this with plots using the <code>patchwork</code> package in R.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="visualizing-taxonomic-data.html#cb100-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;patchwork&quot;</span>)</span>
<span id="cb100-2"><a href="visualizing-taxonomic-data.html#cb100-2" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb100-3"><a href="visualizing-taxonomic-data.html#cb100-3" tabindex="-1"></a></span>
<span id="cb100-4"><a href="visualizing-taxonomic-data.html#cb100-4" tabindex="-1"></a><span class="fu">wrap_plots</span>(birds_richness, sediments_richness, <span class="at">nrow=</span><span class="dv">1</span>)</span></code></pre></div>
Your final box plot should look like this:
<div class="figure"><span style="display:block;" id="fig:fig5"></span>
<img src="images/boxplots.png" alt="Raw version of the taxa boxplot showing alpha-diversity metrics for seabird fecal samples (left) and sediment samples (right)"  />
<p class="caption">
Figure 9.5: Raw version of the taxa boxplot showing alpha-diversity metrics for seabird fecal samples (left) and sediment samples (right)
</p>
</div>
<p>If you want to use other diversity metrics for a plot like this, Phyloseq offers all of the following metrics:</p>
<ul>
<li><p><code>Observed</code>: The number of distinct OTUs (Operational Taxonomic Units) or taxa observed in a sample, reflecting species richness.</p></li>
<li><p><code>Chao1</code>: An estimate of total species richness, taking into account both observed species and the number of rare or singleton species. It provides a more robust estimate of richness.</p></li>
<li><p><code>ACE</code> (Abundance-based Coverage Estimator): Similar to Chao1, it estimates total species richness while considering the abundance distribution of rare species. It is often used in cases where there are many rare species.</p></li>
<li><p><code>Shannon</code>: Measures both richness and evenness of species in a sample. Higher values indicate higher diversity, considering both species number and their relative abundances.</p></li>
<li><p><code>Simpson</code>: Focuses on the dominance of a few species in a sample. Lower values indicate higher diversity, with less dominance by a single species.</p></li>
<li><p><code>InvSimpson</code> (Inverse Simpson): Similar to Simpson but transformed to emphasize the dominance of species. Higher values indicate higher diversity.</p></li>
<li><p><code>Fisher</code>: A rarefaction-based diversity index that estimates species richness while considering the number of individuals sampled. It is sensitive to rare species and is used to compare diversity between samples with different sampling efforts.</p></li>
</ul>
<hr />
 
<hr />
<p style="text-align: center;">
Github: <a href="https://github.com/johannabosch">johannabosch</a> </a>
</p>
<p style="text-align: center;">
<span style="color: #808080;"><em><a href="mailto:yohannabosch@gmail.com" class="email">yohannabosch@gmail.com</a></em></span>
</p>
<!-- Add icon library -->
<p><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"></p>
<!-- Add font awesome icons -->
<p style="text-align: center;">
<a href="https://www.instagram.com/yohannabosch/" class="fa fa-instagram"></a>
<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
<a href="https://github.com/johannabosch/" class="fa fa-github"></a>
</p>
<p> </p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="assigning-taxonomy.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="resources.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/09-VisualizingData.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
